<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta name="google-site-verification" content=""> <meta name="msvalidate.01" content=""> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title>Publications | BioPIM</title> <meta name="author" content="BioPIM "> <meta name="description" content=""> <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous"> <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="/assets/img/favicon.jpg"> <link rel="stylesheet" href="/assets/css/main.css"> <link rel="canonical" href="http://www.biopim.eu/publications/"> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"><span class="font-weight-bold"><img src="/assets/img/biopim_toplogo_small.png"> </span> </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">Home</a> </li> <li class="nav-item "> <a class="nav-link" href="/partners/">Partners</a> </li> <li class="nav-item "> <a class="nav-link" href="/advisors/">Advisors</a> </li> <li class="nav-item "> <a class="nav-link" href="/wp/">WPs</a> </li> <li class="nav-item "> <a class="nav-link" href="/training/">Training</a> </li> <li class="nav-item active"> <a class="nav-link" href="/publications/">Publications<span class="sr-only">(current)</span></a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">Repositories</a> </li> <li class="nav-item "> <a class="nav-link" href="/positions/">Positions</a> </li> <li class="nav-item "> <a class="nav-link" href="/news">News</a> </li> <li class="nav-item "> <a class="nav-link" href="/events">Events</a> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5"> <div class="post"> <header class="post-header"> <h1 class="post-title">Publications</h1> <p class="post-description"></p> </header> <article> <div class="publications"> <h2 class="year">2024</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">IEEE TBCS</abbr></div> <div id="harary2024gcoc" class="col-sm-8"> <div class="title">GCOC: A genome classifier-on-chip based on similarity search content addressable memory</div> <div class="author"> Yuval Harary, Paz Snapir, Shir Siman Tov, Chen Kruphman, Eyal Rechef, <a href="https://www.linkedin.com/in/zuher-jahshan-7a7199196/" rel="external nofollow noopener" target="_blank">Zuher Jahshan</a>, Esteban Garzon, and <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>IEEE Transactions on Biomedical Circuits and Systems</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://ieeexplore.ieee.org/document/10654290" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>GCOC is a genome classification system-on-chip (SoC) that classifies genomes by k-mer matching, an approach that divides a DNA query sequence into a set of short DNA fragments of size k, which are searched in a reference genome database, with the underlying assumption that sequenced DNA reads of the same organism (or its close variants) share most of such k-mers. At the core of GCOC is a similarity, or approximate search-capable Content Addressable Memory (SAS-CAM), which in addition to exact match, also supports approximate, or Hamming distance tolerant search. Classification operation is controlled by an embedded RISC-V processor. GCOC classification platform was designed and manufactured in a commercial 65nm process. We conduct a thorough analysis of GCOC classification efficiency as well as its performance, silicon area, and power consumption using silicon measurements. GCOC classifies 769.2K short DNA reads/sec. The silicon area of GCOC SoC is 3.12mm2 and its power consumption is 1.27mW. We envision GCOC deployed as a field (for example at points of care) portable classifier where the classification is required to be real-time, easy to operate and energy efficient.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">harary2024gcoc</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{GCOC: A genome classifier-on-chip based on similarity search content addressable memory}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Harary, Yuval and Snapir, Paz and Tov, Shir Siman and Kruphman, Chen and Rechef, Eyal and Jahshan, Zuher and Garzon, Esteban and Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{IEEE Transactions on Biomedical Circuits and Systems}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{IEEE}</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">IEEEXplore</abbr></div> <div id="demoor:hal-04821180" class="col-sm-8"> <div class="title">MiMyCS: A Processing-in-Memory Read Mapper for Compressing Next-Gen Sequencing Datasets</div> <div class="author"> Florestan Moor, Meven Mognol, Charles Deltel, Erwan Drezen, Julien Legriel, and <a href="https://lavenier.net/homepage/" rel="external nofollow noopener" target="_blank">Dominique Lavenier</a> </div> <div class="periodical"> <em></em> Dec 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://inria.hal.science/hal-04821180" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://inria.hal.science/hal-04821180v1/file/BIBM2024_paper.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>As Next-Gen sequencing (NGS) technologies keep improving their accuracy and get largely deployed in human health care infrastructures, it is critical to design efficient reference-based compressors that fully leverage the capabilities of modern processors and hardware accelerators. This work proposes MiMyCS: a C++ software to achieve Mapping in Memory for Compressing Short reads. It performs lossless reference-based compression of NGS datasets such as Illumina reads. To this end, MiMyCS computes a non-exhaustive mapping against a reference genome and accelerates this step with the Processing-in-Memory architecture developed by the UPMEM company. Such architecture extends the computational power of a machine by adding dual in-line memory modules on which each memory bank has its own processing unit that runs up to 16 threads. This creates a massively parallel environment, well-fitted to alleviate memory bottlenecks. To reduce the overall amount of sequence comparisons and accelerate further the process, MiMyCS also incorporates a Bloom filters-based dispatcher that predicts against which genome parts reads are most likely to be mapped. We show with real whole human sequencing datasets that MiMyCS is able to achieve a speed-up between 1.2x and 2.7x compared to Genozip, the current leading state-of-the-art compressor, while maintaining a comparable compression ratio and lowering the overall energy consumption. The code of MiMyCS is available at https://gitlab.inria.fr/pim/org.pim.srm.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">demoor:hal-04821180</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{MiMyCS: A Processing-in-Memory Read Mapper for Compressing Next-Gen Sequencing Datasets}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{de Moor, Florestan and Mognol, Meven and Deltel, Charles and Drezen, Erwan and Legriel, Julien and Lavenier, Dominique}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">dec</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{BIBM2024}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{1}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{7176}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">IEEE Access</abbr></div> <div id="rawalign2024" class="col-sm-8"> <div class="title">RawAlign: Accurate, Fast, and Scalable Raw Nanopore Signal Mapping via Combining Seeding and Alignment</div> <div class="author"> Joël Lindegger, <a href="https://cfirtina.com" rel="external nofollow noopener" target="_blank">Can Firtina</a>, <a href="https://scholar.google.ch/citations?user=lMuBESUAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Nika Mansouri Ghiasi</a>, Mohammad Sadrosadati, <a href="https://mealser.github.io" rel="external nofollow noopener" target="_blank">Mohammed Alser</a>, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em>IEEE Access</em>, Dec 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://arxiv.org/abs/2310.05037" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://arxiv.org/pdf/2310.05037" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1109/ACCESS.2024.3520669" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1109/ACCESS.2024.3520669" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Nanopore sequencers generate raw electrical signals representing the contents of a biological sequence molecule passing through the nanopore. These signals can be analyzed directly, avoiding basecalling entirely. We observe that while existing proposals for raw signal analysis typically do well in all metrics for small genomes (e.g., viral genomes), they all perform poorly for large genomes (e.g., the human genome). Our goal is to analyze raw nanopore signals in an accurate, fast, and scalable manner. To this end, we propose RawAlign, the first work to integrate fine-grained signal alignment into the state-of-the-art raw signal mapper. To enable accurate, fast, and scalable mapping with alignment, RawAlign implements three algorithmic improvements and hardware acceleration via a vectorized implementation of fine-grained alignment. Together, these significantly reduce the overhead of typically computationally expensive fine-grained alignment. Our extensive evaluations on different use cases and various datasets show RawAlign provides 1) the most accurate mapping for large genomes and 2) and on-par performance compared to RawHash (between 0.80x-1.08x), while achieving better performance than UNCALLED and Sigmap by on average (geo. mean) 2.83x and 2.06x, respectively.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">rawalign2024</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{RawAlign: Accurate, Fast, and Scalable Raw Nanopore Signal Mapping via Combining Seeding and Alignment}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lindegger, Joël and Firtina, Can and Ghiasi, Nika Mansouri and Sadrosadati, Mohammad and Alser, Mohammed and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">dec</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{IEEE Access}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{12}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{196855-196865}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Genomics;Nanobioscience;Dynamic programming;Accuracy;Sequential analysis;Heuristic algorithms;Costs;Real-time systems;Single instruction multiple data;Signal mapping;Bioinformatics;genome sequence analysis;read mapping;raw signal analysis;nanopore sequencing}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1109/ACCESS.2024.3520669}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">IEEE VLSI</abbr></div> <div id="taco-precharge-free2024" class="col-sm-8"> <div class="title">Designing Precharge-Free Energy-Efficient Content-Addressable Memories</div> <div class="author"> Ramiro Taco, Esteban Garzón, Robert Hanhan, Adam Teman, <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a>, and Marco Lanuzza</div> <div class="periodical"> <em>IEEE Transactions on Very Large Scale Integration (VLSI) Systems</em>, Oct 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://ieeexplore.ieee.org/abstract/document/10723100" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&amp;arnumber=10723100" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1109/TVLSI.2024.3475036" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1109/TVLSI.2024.3475036" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Content-addressable memory (CAM) is a specialized type of memory that facilitates massively parallel comparison of a search pattern against its entire content. State-of-the-art (SOTA) CAM solutions are either fast but power-hungry (NOR CAM) or slow while consuming less power (nand CAM). These limitations stem from the dynamic precharge operation, leading to excessive power consumption in NOR CAMs and charge-sharing issues in NAND CAMs. In this work, we propose a precharge-free CAM (PCAM) class for energy-efficient applications. By avoiding precharge operation, PCAM consumes less energy than a NAND CAM, while achieving search speed comparable to a NOR CAM. PCAM was designed using a 65-nm CMOS technology and comprehensively evaluated under extensive Monte Carlo (MC) simulations while taking into account layout parasitics. When benchmarked against conventional NAND CAM, PCAM demonstrates improved search run time (reduced by more than 30%) and 15% less search energy. Moreover, PCAM can cut energy consumption by more than 75% when compared to conventional NOR CAM. We further extend our analysis to the application level, functionally evaluating the CAM designs as a fully associative cache using a CPU simulator running various benchmark workloads. This analysis confirms that PCAMs represent an optimal energy-performance design choice for associative memories and their broad spectrum of applications.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">taco-precharge-free2024</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Designing Precharge-Free Energy-Efficient Content-Addressable Memories}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Taco, Ramiro and Garzón, Esteban and Hanhan, Robert and Teman, Adam and Yavits, Leonid and Lanuzza, Marco}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">oct</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{IEEE Transactions on Very Large Scale Integration (VLSI) Systems}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{32}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{12}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{2303-2314}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Transistors;Random access memory;Power demand;Discharges (electric);Capacitance;Memory management;Microprocessors;Energy consumption;Pattern matching;Europe;Content-addressable memory (CAM);fully associative cache;low-power;NAND CAM;NOR CAM;precharge-free}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1109/TVLSI.2024.3475036}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">ICPP</abbr></div> <div id="mognol_2024_parallel_proc" class="col-sm-8"> <div class="title">Parallelization of the Banded Needleman &amp; Wunsch Algorithm on UPMEM PiM Architecture for Long DNA Sequence Alignment</div> <div class="author"> Meven Mognol, <a href="https://lavenier.net/homepage/" rel="external nofollow noopener" target="_blank">Dominique Lavenier</a>, and Julien Legriel</div> <div class="periodical"> <em>In Proceedings of the 53rd International Conference on Parallel Processing</em>, Aug 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.org/10.1145/3673038.3673094" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://dl.acm.org/doi/pdf/10.1145/3673038.3673094" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1145/3673038.3673094" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1145/3673038.3673094" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Sequence alignment is an ubiquitous task in genomics analysis whose performance can be affected by the memory-wall on a processor-centric architecture. Processing-in-Memory (PiM) architectures provide a memory system with integrated computing capabilities to alleviate this bottleneck. In this paper, we present a long read optimized version of the Needleman and Wunsch (N&amp;W) algorithm based on PiM devices developed by the UPMEM company. Such memories add 128 computing units to a standard 16GB DIMM. On a server equipped with 20 UPMEM PiMs, our N&amp;W implementation outperforms standard alignment tools by an order of magnitude compared to a traditional multicore server. Code available at https://github.com/upmem/usecase_dpu_alignment.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">mognol_2024_parallel_proc</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Parallelization of the Banded Needleman \&amp; Wunsch Algorithm on UPMEM PiM Architecture for Long DNA Sequence Alignment}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Mognol, Meven and Lavenier, Dominique and Legriel, Julien}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">aug</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Association for Computing Machinery}</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">{New York, NY, USA}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1145/3673038.3673094}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 53rd International Conference on Parallel Processing}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{1062–1071}</span><span class="p">,</span>
  <span class="na">numpages</span> <span class="p">=</span> <span class="s">{10}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Processing-in-Memory, long read Alignment, parallelism}</span><span class="p">,</span>
  <span class="na">location</span> <span class="p">=</span> <span class="s">{Gotland, Sweden}</span><span class="p">,</span>
  <span class="na">series</span> <span class="p">=</span> <span class="s">{ICPP '24}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">ISCA</abbr></div> <div id="10609714" class="col-sm-8"> <div class="title">QUETZAL: Vector Acceleration Framework for Modern Genome Sequence Analysis Algorithms</div> <div class="author"> Julian Pavon, Ivan Vargas Valdivieso, Carlos Rojas, Cesar Hernandez, Mehmet Aslan, Roger Figueras, Yichao Yuan, Joël Lindegger, <a href="https://mealser.github.io" rel="external nofollow noopener" target="_blank">Mohammed Alser</a>, Francesc Moll, Santiago Marco-Sola, Oguz Ergin, Nishil Talati, <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a>, Osman Unsal, Mateo Valero, and Adrian Cristal</div> <div class="periodical"> <em></em> Jun 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://ieeexplore.ieee.org/document/10609714" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://people.inf.ethz.ch/omutlu/pub/QUETZAL_isca24.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1109/ISCA59077.2024.00050" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1109/ISCA59077.2024.00050" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Genome sequence analysis is fundamental to medical breakthroughs such as developing vaccines, enabling genome editing, and facilitating personalized medicine. The exponentially expanding sequencing datasets and complexity of sequencing algorithms necessitate performance enhancements. While the performance of software solutions is constrained by their underlying hardware platforms, the utility of fixed-function accelerators is restricted to only certain sequencing algorithms.This paper presents QUETZAL, the first general-purpose vector acceleration framework designed for high efficiency and broad applicability across a diverse set of genomics algorithms. While a commercial CPU’s vector datapath is a promising candidate to exploit the data-level parallelism in genomics algorithms, our analysis finds that its performance is often limited due to long-latency scatter/gather memory instructions. QUETZAL introduces a hardware-software co-design comprising an accelerator microarchitecture closely integrated with the CPU’s vector datapath, alongside novel vector instructions to fully capitalize on the proposed hardware. QUETZAL integrates a set of scratchpad-style buffers meticulously designed to minimize latency associated with scatter/gather instructions during the retrieval of input genome sequences data. QUETZAL supports both short and long reads, and different types of sequencing data formats. A combination of hardware and software techniques enables QUETZAL to reduce the latency of memory instructions, perform complex computation using a single instruction, and transform data representations at runtime, resulting in overall efficiency gain. QUETZAL significantly accelerates a vectorized CPU baseline on modern genome sequence analysis algorithms by 5.7×, while incurring a small area overhead of 1.4% post place-and-route at the 7nm technology node compared to an HPC ARM CPU.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">10609714</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{QUETZAL: Vector Acceleration Framework for Modern Genome Sequence Analysis Algorithms}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Pavon, Julian and Valdivieso, Ivan Vargas and Rojas, Carlos and Hernandez, Cesar and Aslan, Mehmet and Figueras, Roger and Yuan, Yichao and Lindegger, Joël and Alser, Mohammed and Moll, Francesc and Marco-Sola, Santiago and Ergin, Oguz and Talati, Nishil and Mutlu, Onur and Unsal, Osman and Valero, Mateo and Cristal, Adrian}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{2024 ACM/IEEE 51st Annual International Symposium on Computer Architecture (ISCA)}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jun</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{597-612}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Sequential analysis;Sequences;Software algorithms;Genomics;Approximation algorithms;Vectors;Hardware;Vector architecture;Genome Sequence Analysis;Flexible domain specific accelerator;CPU microarchitecture;Vector acceleration framework;Scratchpad memory}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1109/ISCA59077.2024.00050}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">ISCA</abbr></div> <div id="10609570" class="col-sm-8"> <div class="title">MegIS: High-Performance, Energy-Efficient, and Low-Cost Metagenomic Analysis with In-Storage Processing</div> <div class="author"> <a href="https://scholar.google.ch/citations?user=lMuBESUAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Nika Mansouri Ghiasi</a>, Mohammad Sadrosadati, Harun Mustafa, Arvid Gollwitzer, <a href="https://cfirtina.com" rel="external nofollow noopener" target="_blank">Can Firtina</a>, Julien Eudine, <a href="https://hybol1993.github.io/" rel="external nofollow noopener" target="_blank">Haiyu Mao</a>, Joël Lindegger, <a href="https://scholar.google.com/citations?user=vrbq_BkAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Meryem Banu Cavlak</a>, <a href="https://mealser.github.io" rel="external nofollow noopener" target="_blank">Mohammed Alser</a>, Jisung Park, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em></em> Jun 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://ieeexplore.ieee.org/abstract/document/10609570" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&amp;arnumber=10609570" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Metagenomics, the study of the genome sequences of diverse organisms in a common environment, has led to significant advances in many fields. Since the species present in a metagenomic sample are not known in advance, metagenomic analysis commonly involves the key tasks of determining the species present in a sample and their relative abundances. These tasks require searching large metagenomic databases containing information on different species’ genomes. Metagenomic analysis suffers from significant data movement overhead due to moving large amounts of low-reuse data from the storage system to the rest of the system. In-storage processing can be a fundamental solution for reducing this overhead. However, designing an in-storage processing system for metagenomics is challenging because existing approaches to metagenomic analysis cannot be directly implemented in storage effectively due to the hardware limitations of modern SSDs.We propose MegIS, the first in-storage processing system designed to significantly reduce the data movement overhead of the end-to-end metagenomic analysis pipeline. MegIS is enabled by our lightweight design that effectively leverages and orchestrates processing inside and outside the storage system. Through our detailed analysis of the end-to-end metagenomic analysis pipeline and careful hardware/software co-design, we address in-storage processing challenges for metagenomics via specialized and efficient 1) task partitioning, 2) data/computation flow coordination, 3) storage technology-aware algorithmic optimizations, 4) data mapping, and 5) lightweight in-storage accelerators. MegIS’s design is flexible, capable of supporting different types of metagenomic input datasets, and can be integrated into various metagenomic analysis pipelines. Our evaluation shows that MegIS outperforms the state-of-the-art performance- and accuracy-optimized software metagenomic tools by 2.7× – 37.2× and 6.9×–100.2×, respectively, while matching the accuracy of the accuracy-optimized tool. MegIS achieves 1.5×–5.1× speedup compared to the state-of-the-art metagenomic hardware-accelerated (using processing-in-memory) tool, while achieving significantly higher accuracy.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">10609570</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{MegIS: High-Performance, Energy-Efficient, and Low-Cost Metagenomic Analysis with In-Storage Processing}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Ghiasi, Nika Mansouri and Sadrosadati, Mohammad and Mustafa, Harun and Gollwitzer, Arvid and Firtina, Can and Eudine, Julien and Mao, Haiyu and Lindegger, Joël and Cavlak, Meryem Banu and Alser, Mohammed and Park, Jisung and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jun</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{2024 ACM/IEEE 51st Annual International Symposium on Computer Architecture (ISCA)}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{660-677}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">LNCS</abbr></div> <div id="yann2024_energy_review" class="col-sm-8"> <div class="title">Energy Efficiency Impact of Processing in Memory: A Comprehensive Review of Workloads on the UPMEM Architecture</div> <div class="author"> Yann Falevoz, and Julien Legriel</div> <div class="periodical"> <em></em> Apr 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://link.springer.com/chapter/10.1007/978-3-031-48803-0_13" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1007/978-3-031-48803-0_13" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1007/978-3-031-48803-0_13" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Processing-in-Memory (PIM) architectures have emerged as a promising solution for data-intensive applications, providing significant speedup by processing data directly within the memory. However, the impact of PIM on energy efficiency is not well characterized. In this paper, we provide a comprehensive review of workloads ported to the first PIM product available on the market, namely the UPMEM architecture, and quantify the impact on each workload in terms of energy efficiency. Less than the half of the reviewed papers provide insights on the impact of PIM on energy efficiency, and the evaluation methods differ from one paper to the other. To provide a comprehensive overview, we propose a methodology for estimating energy consumption and efficiency for both the PIM and baseline systems at data center level, enabling a direct comparison of the two systems. Our results show that PIM can provide significant energy savings for data intensive workloads. We also identify key factors that impact the energy efficiency of UPMEM PIM, including the workload characteristics. Overall, this paper provides valuable insights for researchers and practitioners looking to optimize energy efficiency in data-intensive applications using UPMEM PIM architecture.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">yann2024_energy_review</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Energy Efficiency Impact of Processing in Memory: A Comprehensive Review of Workloads on the UPMEM Architecture}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Falevoz, Yann and Legriel, Julien}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">apr</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Euro-Par 2023: Parallel Processing Workshops}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{155-166}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1007/978-3-031-48803-0_13}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">IEEE CAL</abbr></div> <div id="jahshan2024_majork" class="col-sm-8"> <div class="title">MajorK: Majority Based kmer Matching in Commodity DRAM</div> <div class="author"> <a href="https://www.linkedin.com/in/zuher-jahshan-7a7199196/" rel="external nofollow noopener" target="_blank">Zuher Jahshan</a>, and <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>IEEE Computer Architecture Letters</em>, Apr 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://ieeexplore.ieee.org/abstract/document/10488669" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Fast parallel search capabilities on large datasets are required across multiple application domains. One such domain is genome analysis, which requires high-performance k mer matching in large genome databases. Recently proposed solutions implemented k mer matching in DRAM, utilizing its sheer capacity and parallelism. However, their operation is essentially bit-serial, which ultimately limits the performance, especially when matching long strings, as customary in genome analysis pipelines. The proposed solution, MajorK, enables bit-parallel majority based k mer matching in an unmodified commodity DRAM. MajorK employs multiple DRAM row activation, where the search patterns (query k mers) are coded into DRAM addresses. We evaluate MajorK on viral genome k mer matching and show that it can achieve up to 2.7 \times higher performance while providing a better matching accuracy compared to state-of-the-art DRAM based k mer matching accelerators.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">jahshan2024_majork</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{MajorK: Majority Based kmer Matching in Commodity DRAM}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Jahshan, Zuher and Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">apr</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{IEEE Computer Architecture Letters}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">IEEE Access</abbr></div> <div id="jahshan2024_viRal" class="col-sm-8"> <div class="title">ViRAL: Vision Transformer Based Accelerator for ReAL Time Lineage Assignment of Viral Pathogens</div> <div class="author"> <a href="https://www.linkedin.com/in/zuher-jahshan-7a7199196/" rel="external nofollow noopener" target="_blank">Zuher Jahshan</a>, Esteban Garzón, and <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>IEEE Access</em>, Feb 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://ieeexplore.ieee.org/abstract/document/10440271" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&amp;arnumber=10440271" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Real-time genome detection, classification and lineage assignment are critical for efficient tracking of emerging mutations and variants during viral pandemics such as Covid-19. For genomic surveillance to work effectively, each new viral genome sequence must be quickly and accurately associated with an existing viral family (lineage). ViRAL is a hardware-accelerated platform for real-time viral genome lineage assignment based on minhashing and Vision Transformer. Minhashing is a locality sensitive hashing based technique for finding regions of similarity within sequenced genomes. Vision Transformer is a model for image classification that employs a Transformer-like architecture over patches of images. In ViRAL, such image patches are genome fragments extracted from the regions of high similarity. ViRAL is especially efficient in lineage assignment of extremely low quality (or highly ambiguous) genomic data, i.e. when a large fraction of DNA bases are missing in an assembled genome. We implement ViRAL on CPU, GPU and a custom-designed hardware accelerator denoted ACMI. ViRAL assigns newly sequenced SARS-CoV-2 genomes to existing lineages with the top-1 accuracy of 94.2%. The probability of the correct assignment to be found among the five most likely placements generated by ViRAL (top-5 accuracy) is 99.8%. Accelerated ViRAL outperforms the fastest state-of-the-art assignment tools by 69.4\times . It also outperforms ViRAL GPU implementation by 19.5\times . ViRAL strongly outperforms the state-of-the-art solutions in assigning highly-ambiguous genomes: while state-of-the-art tools fail to assign lineage to genomes with 50% ambiguity, ViRAL achieves 77.6% assignment accuracy. We make ViRAL available to the research community through GitHub.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">jahshan2024_viRal</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{ViRAL: Vision Transformer Based Accelerator for ReAL Time Lineage Assignment of Viral Pathogens}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Jahshan, Zuher and Garzón, Esteban and Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">feb</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{IEEE Access}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">Bioinformatics</abbr></div> <div id="jahshan2024_viTal" class="col-sm-8"> <div class="title">ViTAL: Vision TrAnsformer based Low coverage SARS-CoV-2 lineage assignment</div> <div class="author"> <a href="https://www.linkedin.com/in/zuher-jahshan-7a7199196/" rel="external nofollow noopener" target="_blank">Zuher Jahshan</a>, and <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>Bioinformatics</em>, Feb 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://academic.oup.com/bioinformatics/article/40/3/btae093/7610881" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://academic.oup.com/bioinformatics/article-pdf/40/3/btae093/56832253/btae093.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Rapid spread of viral diseases such as Coronavirus disease 2019 (COVID-19) highlights an urgent need for efficient surveillance of virus mutation and transmission dynamics, which requires fast, inexpensive and accurate viral lineage assignment. The first two goals might be achieved through low-coverage whole-genome sequencing (LC-WGS) which enables rapid genome sequencing at scale and at reduced costs. Unfortunately, LC-WGS significantly diminishes the genomic details, rendering accurate lineage assignment very challenging. We present ViTAL, a novel deep learning algorithm specifically designed to perform lineage assignment of low coverage-sequenced genomes. ViTAL utilizes a combination of MinHash for genomic feature extraction and Vision Transformer for fine-grain genome classification and lineage assignment. We show that ViTAL outperforms state-of-the-art tools across diverse coverage levels, reaching up to 87.7% lineage assignment accuracy at 1x coverage where state-of-the-art tools such as UShER and Kraken2 achieve the accuracy of 5.4% and 27.4% respectively. ViTAL achieves comparable accuracy results with up to 8x lower coverage than state-of-the-art tools. We explore ViTAL’s ability to identify the lineages of novel genomes, i.e. genomes the Vision Transformer was not trained on. We show how ViTAL can be applied to preliminary phylogenetic placement of novel variants. The data underlying this article are available in https://github.com/zuherJahshan/vital and can be accessed with 10.5281/zenodo.10688110.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">jahshan2024_viTal</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{ViTAL: Vision TrAnsformer based Low coverage SARS-CoV-2 lineage assignment}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Jahshan, Zuher and Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">feb</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Bioinformatics}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">IEEE SSCL</abbr></div> <div id="harary2024_occam" class="col-sm-8"> <div class="title">OCCAM: An Error Oblivious CAM</div> <div class="author"> Yuval Harary, Paz Snapir, Eyal Reshef, Esteban Garzón, and <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>IEEE Solid-State Circuits Letters</em>, Feb 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://ieeexplore.ieee.org/abstract/document/10423391" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&amp;arnumber=10423391" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Content addressable memories (CAMs) are widely used in many applications in general purpose computer microarchitecture, networking and domain-specific hardware accelerators. In addition to storing and reading data, CAMs enable simultaneous compare of query datawords with the entire memory content. Similar to SRAM and DRAM, CAMs are prone to errors and faults. While error correcting codes (ECCs) are widely used in DRAM and SRAM, they are not directly applicable in CAM: if a dataword that is supposed to match a query altered due to an error, it will falsely mismatch even if it is ECC-encoded. We propose OCCAM, an error oblivious CAM, which combines ECC and approximate search (matching) to allow tolerating a large and dynamically configurable number of errors. We manufactured the OCCAM silicon prototype using 65-nm commercial process and verified its error tolerance capabilities through silicon measurements. OCCAM tolerates 11% error rate (7 bit errors in each 64-bit memory row) with 100% sensitivity and specificity.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">harary2024_occam</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{OCCAM: An Error Oblivious CAM}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Harary, Yuval and Snapir, Paz and Reshef, Eyal and Garzón, Esteban and Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">feb</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{IEEE Solid-State Circuits Letters}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">IEEE Access</abbr></div> <div id="esteban2024_fasta" class="col-sm-8"> <div class="title">FASTA: Revisiting Fully Associative Memories in Computer Microarchitecture</div> <div class="author"> Esteban Garzón, Robert Hanhan, Marco Lanuzza, Adam Teman, and <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>IEEE Access</em>, Jan 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://ieeexplore.ieee.org/abstract/document/10409164" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&amp;arnumber=10409164" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Associative access is widely used in fundamental microarchitectural components, such as caches and TLBs. However, associative (or content addressable) memories (CAMs) have been traditionally considered too large, too energy-hungry, and not scalable, and therefore, have limited use in modern computer microarchitecture. This work revisits these presumptions and proposes an energy-efficient fully-associative tag array (FASTA) architecture, based on a novel complementary CAM (CCAM) bitcell. CCAM offers a full CMOS solution for CAM, removing the need for time- and energy-consuming precharge and combining the speed of NOR CAM and low energy consumption of NAND CAM. While providing better performance and energy consumption, CCAM features a larger area compared to state-of-the-art CAM designs. We further show how FASTA can be used to construct a novel aliasing-free, energy-efficient, Very-Many-Way Associative (VMWA) cache. Circuit-level simulations using 16 nm FinFET technology show that a 128 kB FASTA-based 256-way 8-set associative cache is 28% faster and consumes 88% less energy-per-access than a same sized 8-way (256-set) SRAM based cache, while also providing aliasing-free operation. System-level evaluation performed on the Sniper simulator shows that the VMWA cache exhibits lower Misses Per Kilo Instructions (MPKI) for the majority of benchmarks. Specifically, the 256-way associative cache achieves 17.3%, 11.5%, and 1.2% lower average MPKI for L1, L2, and L3 caches, respectively, compared to a 16-way associative cache. The average IPC improvement for L1, L2, and L3 caches are 1.6%, 1.4%, and 0.2%, respectively.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">esteban2024_fasta</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{FASTA: Revisiting Fully Associative Memories in Computer Microarchitecture}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Garzón, Esteban and Hanhan, Robert and Lanuzza, Marco and Teman, Adam and Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jan</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{IEEE Access}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> </ol> <h2 class="year">2023</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">arXiv</abbr></div> <div id="yavits2023drama" class="col-sm-8"> <div class="title">DRAMA: Commodity DRAM based Content Addressable Memory</div> <div class="author"> <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>arXiv</em>, Dec 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://arxiv.org/abs/2312.15527" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://arxiv.org/pdf/2312.15527.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Fast parallel search capabilities on large datasets provided by content addressable memories (CAM) are required across multiple application domains. However compared to RAM, CAMs feature high area overhead and power consumption, and as a result, they scale poorly. The proposed solution, DRAMA, enables CAM, ternary CAM (TCAM) and approximate (similarity) search CAM functionalities in unmodified commodity DRAM. DRAMA performs compare operation in a bit-serial fashion, where the search pattern (query) is coded in DRAM addresses. A single bit compare (XNOR) in DRAMA is identical to a regular DRAM read. AND and OR operations required for NAND CAM and NOR CAM respectively are implemented using nonstandard DRAM timing. We evaluate DRAMA on bacterial DNA classification and show that DRAMA can achieve 3.6 times higher performance and 19.6 times lower power consumption compared to state-of-the-art CMOS CAM based genome classification accelerator.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">yavits2023drama</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{DRAMA: Commodity DRAM based Content Addressable Memory}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">dec</span><span class="p">,</span>
  <span class="na">eprint</span> <span class="p">=</span> <span class="s">{2312.15527}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{arXiv}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">Genome Biology</abbr></div> <div id="singh2023_DL_basecaller" class="col-sm-8"> <div class="title">A Framework for Designing Efficient Deep Learning-Based Genomic Basecallers</div> <div class="author"> <a href="https://scholar.google.ch/citations?hl=en&amp;user=tb097i4AAAAJ" rel="external nofollow noopener" target="_blank">Gagandeep Singh</a>, <a href="https://mealser.github.io" rel="external nofollow noopener" target="_blank">Mohammed Alser</a>, Alireza Khodamoradi, Kristof Denolf, <a href="https://cfirtina.com" rel="external nofollow noopener" target="_blank">Can Firtina</a>, <a href="https://scholar.google.com/citations?user=vrbq_BkAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Meryem Banu Cavlak</a>, Henk Corporaal, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em>bioRxiv</em>, Nov 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://www.biorxiv.org/content/10.1101/2022.11.20.517297v3.article-info" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://www.biorxiv.org/content/10.1101/2022.11.20.517297v3.full.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1101/2022.11.20.517297" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1101/2022.11.20.517297" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Nanopore sequencing is a widely-used high-throughput genome sequencing technology that can sequence long fragments of a genome. Nanopore sequencing generates noisy electrical signals that need to be converted into a standard string of DNA nucleotide bases using a computational step called basecalling. The performance of basecalling has critical implications for all later steps in genome analysis. Many researchers adopt complex deep learning-based models from the speech recognition domain to perform basecalling without considering the compute demands of such models, which leads to slow, inefficient, and memory-hungry basecallers. Therefore, there is a need to reduce the computation and memory cost of basecalling while maintaining accuracy. However, developing a very fast basecaller that can provide high accuracy requires a deep understanding of genome sequencing, machine learning, and hardware design. Our goal is to develop a comprehensive framework for creating deep learning-based basecallers that provide high efficiency and performance. We introduce RUBICON, a framework to develop hardware-optimized basecallers. RUBICON consists of two novel machine-learning techniques that are specifically designed for basecalling. First, we introduce the first quantization-aware basecalling neural architecture search (QABAS) framework to specialize the basecalling neural network architecture for a given hardware acceleration platform while jointly exploring and finding the best bit-width precision for each neural network layer. Second, we develop SkipClip, the first technique to remove the skip connections present in modern basecallers to greatly reduce resource and storage requirements without any loss in basecalling accuracy. We demonstrate the benefits of RUBICON by developing RUBICALL, the first hardware-optimized basecaller that performs fast and accurate basecalling. Our experimental results on state-of-the-art computing systems show that RUBICALL is a fast, memory-efficient, and hardware-friendly basecaller. Compared to the fastest state-of-the-art basecaller, RUBICALL provides a 3.96× speedup with 2.97% higher accuracy. Compared to an expert-designed basecaller, RUBICALL provides a 141.15× speedup without losing accuracy while also achieving a 6.88× and 2.94× reduction in neural network model size and the number of parameters, respectively. We show that RUBICON helps researchers develop hardware-optimized basecallers that are superior to expert-designed models and can inspire independent future ideas.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">singh2023_DL_basecaller</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Singh, Gagandeep and Alser, Mohammed and Khodamoradi, Alireza and Denolf, Kristof and Firtina, Can and Cavlak, Meryem Banu and Corporaal, Henk and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{bioRxiv}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{A Framework for Designing Efficient Deep Learning-Based Genomic Basecallers}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">nov</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1101/2022.11.20.517297}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">PACT</abbr></div> <div id="jinfan2023_simplePIM" class="col-sm-8"> <div class="title">SimplePIM: A Software Framework for Productive and Efficient Processing-in-Memory</div> <div class="author"> Jinfan Chen, <a href="https://ee.ethz.ch/the-department/people-a-z/person-detail.MjQxMjgy.TGlzdC8zMjc5LC0xNjUwNTg5ODIw.html" rel="external nofollow noopener" target="_blank">Juan Gómez-Luna</a>, Izzat El Hajj, Yuxin Guo, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em>Parallel Architectures and Compilation Techniques</em>, Oct 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://arxiv.org/abs/2310.01893" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://arxiv.org/pdf/2310.01893" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Data movement between memory and processors is a major bottleneck in modern computing systems. The processing-in-memory (PIM) paradigm aims to alleviate this bottleneck by performing computation inside memory chips. Real PIM hardware (e.g., the UPMEM system) is now available and has demonstrated potential in many applications. However, programming such real PIM hardware remains a challenge for many programmers. This paper presents a new software framework, SimplePIM, to aid programming real PIM systems. The framework processes arrays of arbitrary elements on a PIM device by calling iterator functions from the host and provides primitives for communication among PIM cores and between PIM and the host system. We implement SimplePIM for the UPMEM PIM system and evaluate it on six major applications. Our results show that SimplePIM enables 66.5% to 83.1% reduction in lines of code in PIM programs. The resulting code leads to higher performance (between 10% and 37% speedup) than hand-optimized code in three applications and provides comparable performance in three others. SimplePIM is fully and freely available at https://github.com/CMU-SAFARI/SimplePIM</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">jinfan2023_simplePIM</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{SimplePIM: A Software Framework for Productive and Efficient Processing-in-Memory}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Chen, Jinfan and Gómez-Luna, Juan and Hajj, Izzat El and Guo, Yuxin and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">oct</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Parallel Architectures and Compilation Techniques}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">IEEE TCS</abbr></div> <div id="esteban2023_sensing_scheme" class="col-sm-8"> <div class="title">A Low-Complexity Sensing Scheme for Approximate Matching Content-Addressable Memory</div> <div class="author"> Esteban Garzón, Roman Golman, Marco Lanuzza, Adam Teman, and <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>IEEE Transactions on Circuits and Systems II</em>, Oct 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://ieeexplore.ieee.org/abstract/document/10153337" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&amp;arnumber=10153337" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>The need for approximate rather than exact search arises in numerous compare-intensive applications, from networking to computational genomics. This brief presents a novel sensing approach for approximate matching content-addressable memory (CAM) designed to handle large Hamming distances (HDs) between the query pattern and stored data. The proposed matchline sensing scheme (MLSS) employs a replica mechanism and a 12-transistor positive feedback sense amplifier to effectively resolve the approximate match operation. The MLSS was integrated into a 4 kB approximate CAM array and fabricated in a 65 nm CMOS technology. With an overall area footprint of 0.0048 mm2, which includes 512 sense amplifiers and the replica mechanism, the MLSS allows a flexible and dynamic adjustment of the HD tolerance threshold via several design variables. Experimental measurements demonstrate the efficiency of our sensing scheme in tolerating very large HDs with the highest sensitivity.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">esteban2023_sensing_scheme</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{A Low-Complexity Sensing Scheme for Approximate Matching Content-Addressable Memory}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Garzón, Esteban and Golman, Roman and Lanuzza, Marco and Teman, Adam and Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">oct</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{IEEE Transactions on Circuits and Systems II}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">ISPASS</abbr></div> <div id="gomezluna_ispass_2023" class="col-sm-8"> <div class="title">An Experimental Evaluation of Machine Learning Training on a Real Processing-in-Memory System</div> <div class="author"> <a href="https://ee.ethz.ch/the-department/people-a-z/person-detail.MjQxMjgy.TGlzdC8zMjc5LC0xNjUwNTg5ODIw.html" rel="external nofollow noopener" target="_blank">Juan Gómez-Luna</a>, Yuxin Guo, Sylvan Brocard, Julien Legriel, Remy Cimadomo, Geraldo F. Oliveira, <a href="https://scholar.google.ch/citations?hl=en&amp;user=tb097i4AAAAJ" rel="external nofollow noopener" target="_blank">Gagandeep Singh</a>, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em></em> Sep 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://arxiv.org/abs/2207.07886" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://arxiv.org/pdf/2207.07886" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Training machine learning (ML) algorithms is a computationally intensive process, which is frequently memory-bound due to repeatedly accessing large training datasets. As a result, processor-centric systems (e.g., CPU, GPU) suffer from costly data movement between memory units and processing units, which consumes large amounts of energy and execution cycles. Memory-centric computing systems, i.e., with processing-in-memory (PIM) capabilities, can alleviate this data movement bottleneck. Our goal is to understand the potential of modern general-purpose PIM architectures to accelerate ML training. To do so, we (1) implement several representative classic ML algorithms (namely, linear regression, logistic regression, decision tree, K-Means clustering) on a real-world general-purpose PIM architecture, (2) rigorously evaluate and characterize them in terms of accuracy, performance and scaling, and (3) compare to their counterpart implementations on CPU and GPU. Our evaluation on a real memory-centric computing system with more than 2500 PIM cores shows that general-purpose PIM architectures can greatly accelerate memory-bound ML workloads, when the necessary operations and datatypes are natively supported by PIM hardware. For example, our PIM implementation of decision tree is 27× faster than a state-of-the-art CPU version on an 8-core Intel Xeon, and 1.34× faster than a state-of-the-art GPU version on an NVIDIA A100. Our K-Means clustering on PIM is 2.8× and 3.2× than state-of-the-art CPU and GPU versions, respectively. To our knowledge, our work is the first one to evaluate ML training on a real-world PIM architecture. We conclude with key observations, takeaways, and recommendations that can inspire users of ML workloads, programmers of PIM architectures, and hardware designers &amp; architects of future memory-centric computing systems.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">gomezluna_ispass_2023</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{An Experimental Evaluation of Machine Learning Training on a Real Processing-in-Memory System}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Gómez-Luna, Juan and Guo, Yuxin and Brocard, Sylvan and Legriel, Julien and Cimadomo, Remy and Oliveira, Geraldo F. and Singh, Gagandeep and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">sep</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">GR</abbr></div> <div id="ekim2023_mapquik" class="col-sm-8"> <div class="title">Efficient mapping of accurate long reads in minimizer space with mapquik</div> <div class="author"> Baris Ekim, Kristoffer Sahlin, Paul Medvedev, Bonnie Berger, and <a href="http://rayan.chikhi.name/" rel="external nofollow noopener" target="_blank">Rayan Chikhi</a> </div> <div class="periodical"> <em>Genome Research</em>, Jun 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://genome.cshlp.org/content/33/7/1188" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://genome.cshlp.org/content/33/7/1188.full.pdf+html" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>DNA sequencing data continue to progress toward longer reads with increasingly lower sequencing error rates. We focus on the critical problem of mapping, or aligning, low-divergence sequences from long reads (e.g., Pacific Biosciences [PacBio] HiFi) to a reference genome, which poses challenges in terms of accuracy and computational resources when using cutting-edge read mapping approaches that are designed for all types of alignments. A natural idea would be to optimize efficiency with longer seeds to reduce the probability of extraneous matches; however, contiguous exact seeds quickly reach a sensitivity limit. We introduce mapquik, a novel strategy that creates accurate longer seeds by anchoring alignments through matches of k consecutively sampled minimizers (k-min-mers) and only indexing k-min-mers that occur once in the reference genome, thereby unlocking ultrafast mapping while retaining high sensitivity. We show that mapquik significantly accelerates the seeding and chaining steps—fundamental bottlenecks to read mapping—for both the human and maize genomes with Formula sensitivity and near-perfect specificity. On the human genome, for both real and simulated reads, mapquik achieves a Formula speedup over the state-of-the-art tool minimap2, and on the maize genome, mapquik achieves a Formula speedup over minimap2, making mapquik the fastest mapper to date. These accelerations are enabled from not only minimizer-space seeding but also a novel heuristic Formula pseudochaining algorithm, which improves upon the long-standing Formula bound. Minimizer-space computation builds the foundation for achieving real-time analysis of long-read sequencing data.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">ekim2023_mapquik</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Efficient mapping of accurate long reads in minimizer space with mapquik}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Ekim, Baris and Sahlin, Kristoffer and Medvedev, Paul and Berger, Bonnie and Chikhi, Rayan}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jun</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Genome Research}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">IEEE Access</abbr></div> <div id="garzon2023dmtj" class="col-sm-8"> <div class="title">A low-energy DMTJ-based ternary content- addressable memory with reliable sub-nanosecond search operation</div> <div class="author"> Esteban Garzón, <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a>, Giovanni Finocchio, Mario Carpentieri, Adam Teman, and Marco Lanuzza</div> <div class="periodical"> <em>IEEE Access</em>, Feb 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://ieeexplore.ieee.org/document/10045679" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&amp;arnumber=10045679" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1109/ACCESS.2023.3245981" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1109/ACCESS.2023.3245981" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>In this paper, we propose an energy-efficient, reliable, hybrid, 10-transistor/2-Double-Barrier-Magnetic-Tunnel-Junction (10T2DMTJ) non-volatile (NV) ternary content-addressable memory (TCAM) with sub-nanosecond search operation. Our cell design relies on low-energy-demanding MTJs organized in a low-complexity voltage-divider-based circuit along with a simple dynamic logic CMOS matching network, which improves the search reliability. The proposed NV-TCAM was designed in 28 nm FDSOI process and evaluated under exhaustive Monte Carlo simulations. When compared to the best previous proposed NV-TCAMs, our solution achieves lower search error rate (3.8×) and lower write and search energy (–73% and–79%, respectively), while also exhibiting smaller area footprint (–74%). Such benefits are achieved at the expense of reduced search speed.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">garzon2023dmtj</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Garzón, Esteban and Yavits, Leonid and Finocchio, Giovanni and Carpentieri, Mario and Teman, Adam and Lanuzza, Marco}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{IEEE Access}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{A low-energy DMTJ-based ternary content- addressable memory with reliable sub-nanosecond search operation}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{11}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">feb</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{16812-16819}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Computer architecture;Microprocessors;Transistors;Magnetic tunneling;Switches;Integrated circuit reliability;Error analysis;Double-barrier MTJ;non-volatile TCAM (NV-TCAM);energy-efficiency;low-power}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1109/ACCESS.2023.3245981}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">ACM TACO</abbr></div> <div id="firtina_aphmm_2023" class="col-sm-8"> <div class="title">ApHMM: Accelerating Profile Hidden Markov Models for Fast and Energy-Efficient Genome Analysis</div> <div class="author"> <a href="https://cfirtina.com" rel="external nofollow noopener" target="_blank">Can Firtina</a>, Kamlesh Pillai, <a href="https://in.linkedin.com/in/gurpreet-singh-kalsi-bb75302" rel="external nofollow noopener" target="_blank">Gurpreet S. Kalsi</a>, Bharathwaj Suresh, <a href="https://damlasenolcali.github.io" rel="external nofollow noopener" target="_blank">Damla Senol Cali</a>, <a href="https://scholar.google.com/citations?user=-pOH1s4AAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Jeremie S. Kim</a>, Taha Shahroodi, <a href="https://scholar.google.com/citations?user=vrbq_BkAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Meryem Banu Cavlak</a>, Joël Lindegger, <a href="https://mealser.github.io" rel="external nofollow noopener" target="_blank">Mohammed Alser</a>, Juan Gómez Luna, <a href="https://in.linkedin.com/in/sreenivas-subramoney-i-am-hiring-6564521" rel="external nofollow noopener" target="_blank">Sreenivas Subramoney</a>, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em>ACM Trans. Archit. Code Optim.</em>, Dec 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://dl.acm.org/doi/10.1145/3632950" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://dl.acm.org/doi/pdf/10.1145/3632950" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> <a href="https://github.com/CMU-SAFARI/ApHMM-GPU" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1145/3632950" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1145/3632950" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Profile hidden Markov models (pHMMs) are widely employed in various bioinformatics applications to identify similarities between biological sequences, such as DNA or protein sequences. In pHMMs, sequences are represented as graph structures, where states and edges capture modifications (i.e., insertions, deletions, and substitutions) by assigning probabilities to them. These probabilities are subsequently used to compute the similarity score between a sequence and a pHMM graph. The Baum-Welch algorithm, a prevalent and highly accurate method, utilizes these probabilities to optimize and compute similarity scores. Accurate computation of these probabilities is essential for the correct identification of sequence similarities. However, the Baum-Welch algorithm is computationally intensive, and existing solutions offer either software-only or hardware-only approaches with fixed pHMM designs. When we analyze state-of-the-art works, we identify an urgent need for a flexible, high-performance, and energy-efficient hardware-software co-design to address the major inefficiencies in the Baum-Welch algorithm for pHMMs. We introduce ApHMM, the first flexible acceleration framework designed to significantly reduce both computational and energy overheads associated with the Baum-Welch algorithm for pHMMs. ApHMM employs hardware-software co-design to tackle the major inefficiencies in the Baum-Welch algorithm by 1) designing flexible hardware to accommodate various pHMM designs, 2) exploiting predictable data dependency patterns through on-chip memory with memoization techniques, 3) rapidly filtering out unnecessary computations using a hardware-based filter, and 4) minimizing redundant computations. ApHMM achieves substantial speedups of 15.55x - 260.03x, 1.83x - 5.34x, and 27.97x when compared to CPU, GPU, and FPGA implementations of the Baum-Welch algorithm, respectively. ApHMM outperforms state-of-the-art CPU implementations in three key bioinformatics applications: 1) error correction, 2) protein family search, and 3) multiple sequence alignment, by 1.29x - 59.94x, 1.03x - 1.75x, and 1.03x - 1.95x, respectively, while improving their energy efficiency by 64.24x - 115.46x, 1.75x, 1.96x.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">firtina_aphmm_2023</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{ApHMM: Accelerating Profile Hidden Markov Models for Fast and Energy-Efficient Genome Analysis}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{ACM Trans. Archit. Code Optim.}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Firtina, Can and Pillai, Kamlesh and Kalsi, Gurpreet S. and Suresh, Bharathwaj and Cali, Damla Senol and Kim, Jeremie S. and Shahroodi, Taha and Cavlak, Meryem Banu and Lindegger, Jo\"{e}l and Alser, Mohammed and Luna, Juan G\'{o}mez and Subramoney, Sreenivas and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">issn</span> <span class="p">=</span> <span class="s">{1544-3566}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">dec</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1145/3632950}</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">IEEE TC</abbr></div> <div id="Merlin2023" class="col-sm-8"> <div class="title">DIPER: Detection and Identification of Pathogens using Edit distance-tolerant Resistive CAM</div> <div class="author"> Itay Merlin, Esteban Garzón, Alex Fish, and <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>IEEE Transactions on Computers</em>, Dec 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://www.computer.org/csdl/journal/tc/5555/01/10260684/1QEx9tfzCXC" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&amp;arnumber=10260684" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1109/tc.2023.3315829" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1109/tc.2023.3315829" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>We propose a novel resistive edit distance-tolerant content addressable memory for computational genomics applications, particularly for detection and identification of pathogens of pandemic importance. Unlike state-of-the-art approximate search solutions that tolerate small number of replacements between the query pattern and the stored data, DIPER tolerates insertions and deletions, ubiquitous in genomics. DIPER achieves up to 1.7× higher F1 score for high-quality DNA reads and up to 6.2× higher F1 score for DNA reads with 15% error rate, compared to state-of-the-art DNA classification tool Kraken2. Simulated at 500MHz, DIPER provides 910× average speedup over Kraken2.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">Merlin2023</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Merlin, Itay and Garzón, Esteban and Fish, Alex and Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{IEEE Transactions on Computers}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{DIPER: Detection and Identification of Pathogens using Edit distance-tolerant Resistive CAM}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">issn</span> <span class="p">=</span> <span class="s">{0018-9340}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{1-12}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1109/tc.2023.3315829}</span><span class="p">,</span>
  <span class="na">owner</span> <span class="p">=</span> <span class="s">{calkan}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">MICRO</abbr></div> <div id="shahroodi_swordfish_2023" class="col-sm-8"> <div class="title">Swordfish: A Framework for Evaluating Deep Neural Network-Based Basecalling Using Computation-In-Memory with Non-Ideal Memristors</div> <div class="author"> Taha Shahroodi, <a href="https://scholar.google.ch/citations?hl=en&amp;user=tb097i4AAAAJ" rel="external nofollow noopener" target="_blank">Gagandeep Singh</a>, Mahdi Zahedi, <a href="https://hybol1993.github.io/" rel="external nofollow noopener" target="_blank">Haiyu Mao</a>, <a href="https://scholar.google.com/citations?user=IhARycsAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Joel Lindegger</a>, <a href="https://cfirtina.com" rel="external nofollow noopener" target="_blank">Can Firtina</a>, Stephan Wong, <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a>, and Said Hamdioui</div> <div class="periodical"> <em>In Proceedings of the 56th Annual IEEE/ACM International Symposium on Microarchitecture (MICRO)</em>, Oct 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.org/10.1145/3613424.3614252" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://arxiv.org/pdf/2310.04366" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1145/3613424.3614252" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1145/3613424.3614252" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Basecalling, an essential step in many genome analysis studies, relies on large Deep Neural Network s (DNN s) to achieve high accuracy. Unfortunately, these DNN s are computationally slow and inefficient, leading to considerable delays and resource constraints in the sequence analysis process. A Computation-In-Memory (CIM) architecture using memristors can significantly accelerate the performance of DNN s. However, inherent device non-idealities and architectural limitations of such designs can greatly degrade the basecalling accuracy, which is critical for accurate genome analysis. To facilitate the adoption of memristor-based CIM designs for basecalling, it is important to (1) conduct a comprehensive analysis of potential CIM architectures and (2) develop effective strategies for mitigating the possible adverse effects of inherent device non-idealities and architectural limitations. This paper proposes Swordfish, a novel hardware/software co-design framework that can effectively address the two aforementioned issues. Swordfish incorporates seven circuit and device restrictions or non-idealities from characterized real memristor-based chips. Swordfish leverages various hardware/software co-design solutions to mitigate the basecalling accuracy loss due to such non-idealities. To demonstrate the effectiveness of Swordfish, we take Bonito, the state-of-the-art (i.e., accurate and fast), open-source basecaller as a case study. Our experimental results using Swordfish show that a CIM architecture can realistically accelerate Bonito for a wide range of real datasets by an average of 25.7x, with an accuracy loss of 6.01%.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">shahroodi_swordfish_2023</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Swordfish: A Framework for Evaluating Deep Neural Network-Based Basecalling Using Computation-In-Memory with Non-Ideal Memristors}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 56th Annual IEEE/ACM International Symposium on Microarchitecture (MICRO)}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Shahroodi, Taha and Singh, Gagandeep and Zahedi, Mahdi and Mao, Haiyu and Lindegger, Joel and Firtina, Can and Wong, Stephan and Mutlu, Onur and Hamdioui, Said}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">oct</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{1437–1452}</span><span class="p">,</span>
  <span class="na">isbn</span> <span class="p">=</span> <span class="s">{9798400703294}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1145/3613424.3614252}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">bioRxiv</abbr></div> <div id="Jahshan2023" class="col-sm-8"> <div class="title">DASH-CAM: Dynamic Approximate SearcH Content Addressable Memory for genome classification</div> <div class="author"> <a href="https://www.linkedin.com/in/zuher-jahshan-7a7199196/" rel="external nofollow noopener" target="_blank">Zuher Jahshan</a>, Itay Merlin, Esteban Garzón, and <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>bioRxiv</em>, Oct 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://www.biorxiv.org/content/early/2023/10/02/2023.09.29.560142" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://www.biorxiv.org/content/10.1101/2023.09.29.560142v1.full.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1101/2023.09.29.560142" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1101/2023.09.29.560142" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>We propose a novel dynamic storage-based approximate search content addressable memory (DASH-CAM) for computational genomics applications, particularly for identification and classification of viral pathogens of epidemic significance. DASH-CAM provides 5.5x better density compared to state-of-the-art SRAM-based approximate search CAM. This allows using DASH-CAM as a portable classifier that can be applied to pathogen surveillance in low-quality field settings during pandemics, as well as to pathogen diagnostics at points of care. DASH-CAM approximate search capabilities allow a high level of flexibility when dealing with a variety of industrial sequencers with different error profiles. DASH-CAM achieves up to 30% and 20% higher F1 score when classifying DNA reads with 10% error rate, compared to state-of-the-art DNA classification tools MetaCache-GPU and Kraken2 respectively. Simulated at 1GHz, DASH-CAM provides 1,178x and 1,040x average speedup over MetaCache-GPU and Kraken2 respectively.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">Jahshan2023</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Jahshan, Zuher and Merlin, Itay and Garz{\'o}n, Esteban and Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{DASH-CAM: Dynamic Approximate SearcH Content Addressable Memory for genome classification}</span><span class="p">,</span>
  <span class="na">elocation-id</span> <span class="p">=</span> <span class="s">{2023.09.29.560142}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">oct</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1101/2023.09.29.560142}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Cold Spring Harbor Laboratory}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{bioRxiv}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">arXiv</abbr></div> <div id="firtina_rawhash2_2023" class="col-sm-8"> <div class="title">RawHash2: Accurate and Fast Mapping of Raw Nanopore Signals using a Hash-based Seeding Mechanism</div> <div class="author"> <a href="https://cfirtina.com" rel="external nofollow noopener" target="_blank">Can Firtina</a>, Melina Soysal, Joël Lindegger, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em>arXiv</em>, Sep 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.org/10.48550/arXiv.2309.05771" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://arxiv.org/pdf/2309.05771.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> <a href="https://github.com/CMU-SAFARI/RawHash" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.48550/arXiv.2309.05771" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.48550/arXiv.2309.05771" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Raw nanopore signals can be analyzed while they are being generated, a process known as real-time analysis. Real-time analysis of raw signals is essential to utilize the unique features that nanopore sequencing provides, enabling the early stopping of the sequencing of a read or the entire sequencing run based on the analysis. The state-of-the-art mechanism, RawHash, offers the first hash-based efficient and accurate similarity identification between raw signals and a reference genome by quickly matching their hash values. In this work, we introduce RawHash2, which provides major improvements over RawHash, including a more sensitive chaining implementation, weighted mapping decisions, frequency filters to reduce ambiguous seed hits, minimizers for hash-based sketching, and support for the R10.4 flow cell version and various data formats such as POD5. Compared to RawHash, RawHash2 provides the best F1 accuracy in all datasets and provides better average throughput by 2.5x. Source code is available at https://github.com/CMU-SAFARI/RawHash.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">firtina_rawhash2_2023</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{{RawHash2}: Accurate and Fast Mapping of Raw Nanopore Signals using a Hash-based Seeding Mechanism}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Firtina, Can and Soysal, Melina and Lindegger, Joël and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{arXiv}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">sep</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.48550/arXiv.2309.05771}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">arXiv</abbr></div> <div id="Eudine2023gateseeder" class="col-sm-8"> <div class="title">GateSeeder: Near-memory CPU-FPGA Acceleration of Short and Long Read Mapping</div> <div class="author"> Julien Eudine, <a href="https://mealser.github.io" rel="external nofollow noopener" target="_blank">Mohammed Alser</a>, <a href="https://scholar.google.ch/citations?hl=en&amp;user=tb097i4AAAAJ" rel="external nofollow noopener" target="_blank">Gagandeep Singh</a>, <a href="https://www.alkanlab.org/team/can-alkan/" rel="external nofollow noopener" target="_blank">Can Alkan</a>, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em>arXiv</em>, Sep 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://arxiv.org/abs/2309.17063" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://arxiv.org/pdf/2309.17063" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> <a href="https://github.com/CMU-SAFARI/GateSeeder" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Motivation: Read mapping is a computationally expensive process and a major bottleneck in genomics analyses. The performance of read mapping is mainly limited by the performance of three key computational steps: Index Querying, Seed Chaining, and Sequence Alignment. The first step is dominated by how fast and frequent it accesses the main memory (i.e., memory-bound), while the latter two steps are dominated by how fast the CPU can compute their computationally-costly dynamic programming algorithms (i.e., compute-bound). Accelerating these three steps by exploiting new algorithms and new hardware devices is essential to accelerate most genome analysis pipelines that widely use read mapping. Given the large body of work on accelerating Sequence Alignment, this work focuses on significantly improving the remaining steps. Results: We introduce GateSeeder, the first CPU-FPGA-based near-memory acceleration of both short and long read mapping. GateSeeder exploits near-memory computation capability provided by modern FPGAs that couple a reconfigurable compute fabric with high-bandwidth memory (HBM) to overcome the memory-bound and compute-bound bottlenecks. GateSeeder also introduces a new lightweight algorithm for finding the potential matching segment pairs. Using real ONT, HiFi, and Illumina sequences, we experimentally demonstrate that GateSeeder outperforms Minimap2, without performing sequence alignment, by up to 40.3x, 4.8x, and 2.3x, respectively. When performing read mapping with sequence alignment, GateSeeder outperforms Minimap2 by 1.15-4.33x (using KSW2) and by 1.97-13.63x (using WFA-GPU)</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">Eudine2023gateseeder</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{GateSeeder: Near-memory CPU-FPGA Acceleration of Short and Long Read Mapping}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Eudine, Julien and Alser, Mohammed and Singh, Gagandeep and Alkan, Can and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">sep</span><span class="p">,</span>
  <span class="na">eprint</span> <span class="p">=</span> <span class="s">{2309.17063}</span><span class="p">,</span>
  <span class="na">archiveprefix</span> <span class="p">=</span> <span class="s">{arXiv}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{arXiv}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">IEEE VLSI</abbr></div> <div id="Khalifa2023" class="col-sm-8"> <div class="title">ClaPIM: Scalable Sequence CLAssification using Processing-In-Memory</div> <div class="author"> Marcel Khalifa, Barak Hoffer, Orian Leitersdorf, Robert Hanhan, Ben Perach, <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a>, and Shahar Kvatinsky</div> <div class="periodical"> <em>IEEE Transactions on Very Large Scale Integration (VLSI) Systems</em>, Jul 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://ieeexplore.ieee.org/document/10192521" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://asic2.group/wp-content/uploads/2023/07/ClaPIM_Marcel-Khalifa.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.48550/arXiv.2302.08284" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.48550/arXiv.2302.08284" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>DNA sequence classification is a fundamental task in computational biology with vast implications for applications such as disease prevention and drug design. Therefore, fast high-quality sequence classifiers are significantly important. This paper introduces ClaPIM, a scalable DNA sequence classification architecture based on the emerging concept of hybrid in-crossbar and near-crossbar memristive processing-in-memory (PIM). We enable efficient and high-quality classification by uniting the filter and search stages within a single algorithm. Specifically, we propose a custom filtering technique that drastically narrows the search space and a search approach that facilitates approximate string matching through a distance function. ClaPIM is the first PIM architecture for scalable approximate string matching that benefits from the high density of memristive crossbar arrays and the massive computational parallelism of PIM. Compared with Kraken2, a state-of-the-art software classifier, ClaPIM provides significantly higher classification quality (up to 20x improvement in F1 score) and also demonstrates a 1.8x throughput improvement. Compared with EDAM, a recently-proposed SRAM-based accelerator that is restricted to small datasets, we observe both a 30.4x improvement in normalized throughput per area and a 7% increase in classification precision.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">Khalifa2023</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Khalifa, Marcel and Hoffer, Barak and Leitersdorf, Orian and Hanhan, Robert and Perach, Ben and Yavits, Leonid and Kvatinsky, Shahar}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{IEEE Transactions on Very Large Scale Integration (VLSI) Systems}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{ClaPIM: Scalable Sequence CLAssification using Processing-In-Memory}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{31}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{9}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jul</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{1347--1357}</span><span class="p">,</span>
  <span class="na">archiveprefix</span> <span class="p">=</span> <span class="s">{arXiv}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.48550/arXiv.2302.08284}</span><span class="p">,</span>
  <span class="na">eid</span> <span class="p">=</span> <span class="s">{arXiv:2302.08284}</span><span class="p">,</span>
  <span class="na">eprint</span> <span class="p">=</span> <span class="s">{2302.08284}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Computer Science - Machine Learning, Electrical Engineering and Systems Science - Systems and Control}</span><span class="p">,</span>
  <span class="na">owner</span> <span class="p">=</span> <span class="s">{calkan}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">Bioinformatics</abbr></div> <div id="firtina_rawhash_2023" class="col-sm-8"> <div class="title">RawHash: Enabling Fast and Accurate Real-Time Analysis of Raw Nanopore Signals for Large Genomes</div> <div class="author"> <a href="https://cfirtina.com" rel="external nofollow noopener" target="_blank">Can Firtina</a>, <a href="https://scholar.google.ch/citations?user=lMuBESUAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Nika Mansouri Ghiasi</a>, <a href="https://scholar.google.com/citations?user=IhARycsAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Joel Lindegger</a>, <a href="https://scholar.google.ch/citations?hl=en&amp;user=tb097i4AAAAJ" rel="external nofollow noopener" target="_blank">Gagandeep Singh</a>, <a href="https://scholar.google.com/citations?user=vrbq_BkAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Meryem Banu Cavlak</a>, <a href="https://hybol1993.github.io/" rel="external nofollow noopener" target="_blank">Haiyu Mao</a>, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em>Bioinformatics</em>, Jul 2023 </div> <div class="periodical"> Proceedings of the 31st Annual Conference on Intelligent Systems for Molecular Biology (ISMB) and the 22nd European Conference on Computational Biology (ECCB) </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://arxiv.org/abs/2301.09200" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://arxiv.org/pdf/2301.09200" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> <a href="https://github.com/CMU-SAFARI/RawHash" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1093/bioinformatics/btad272" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1093/bioinformatics/btad272" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Nanopore sequencers generate electrical raw signals in real-time while sequencing long genomic strands. These raw signals can be analyzed as they are generated, providing an opportunity for real-time genome analysis. An important feature of nanopore sequencing, Read Until, can eject strands from sequencers without fully sequencing them, which provides opportunities to computationally reduce the sequencing time and cost. However, existing works utilizing Read Until either 1) require powerful computational resources that may not be available for portable sequencers or 2) lack scalability for large genomes, rendering them inaccurate or ineffective. We propose RawHash, the first mechanism that can accurately and efficiently perform real-time analysis of nanopore raw signals for large genomes using a hash-based similarity search. To enable this, RawHash ensures the signals corresponding to the same DNA content lead to the same hash value, regardless of the slight variations in these signals. RawHash achieves an accurate hash-based similarity search via an effective quantization of the raw signals such that signals corresponding to the same DNA content have the same quantized value and, subsequently, the same hash value. We evaluate RawHash on three applications: 1) read mapping, 2) relative abundance estimation, and 3) contamination analysis. Our evaluations show that RawHash is the only tool that can provide high accuracy and high throughput for analyzing large genomes in real-time. When compared to the state-of-the-art techniques, UNCALLED and Sigmap, RawHash provides 1) 25.8x and 3.4x better average throughput and 2) significantly better accuracy for large genomes, respectively. Source code is available at https://github.com/CMUSAFARI/RawHash.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">firtina_rawhash_2023</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Firtina, Can and Ghiasi, Nika Mansouri and Lindegger, Joel and Singh, Gagandeep and Cavlak, Meryem Banu and Mao, Haiyu and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Bioinformatics}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{{RawHash}: {Enabling} {Fast} and {Accurate} {Real}-{Time} {Analysis} of {Raw} {Nanopore} {Signals} for {Large} {Genomes}}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jul</span><span class="p">,</span>
  <span class="na">note</span> <span class="p">=</span> <span class="s">{Proceedings of the 31st Annual Conference on Intelligent Systems for Molecular Biology (ISMB) and the 22nd European Conference on Computational Biology (ECCB)}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{Supplement 1}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{i297–i307}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{39}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1093/bioinformatics/btad272}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">DAC</abbr></div> <div id="mutlu2023accelerating" class="col-sm-8"> <div class="title">Accelerating Genome Analysis via Algorithm-Architecture Co-Design</div> <div class="author"> <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a>, and <a href="https://cfirtina.com" rel="external nofollow noopener" target="_blank">Can Firtina</a> </div> <div class="periodical"> <em>In Proceedings of the 60th Design Automation Conference (DAC)</em>, Jul 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.org/10.48550/arXiv.2305.00492" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://arxiv.org/pdf/2305.00492" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.48550/arXiv.2305.00492" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.48550/arXiv.2305.00492" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>High-throughput sequencing (HTS) technologies have revolutionized the field of genomics, enabling rapid and cost-effective genome analysis for various applications. However, the increasing volume of genomic data generated by HTS technologies presents significant challenges for computational techniques to effectively analyze genomes. To address these challenges, several algorithm-architecture co-design works have been proposed, targeting different steps of the genome analysis pipeline. These works explore emerging technologies to provide fast, accurate, and low-power genome analysis. This paper provides a brief review of the recent advancements in accelerating genome analysis, covering the opportunities and challenges associated with the acceleration of the key steps of the genome analysis pipeline. Our analysis highlights the importance of integrating multiple steps of genome analysis using suitable architectures to unlock significant performance improvements and reduce data movement and energy consumption. We conclude by emphasizing the need for novel strategies and techniques to address the growing demands of genomic data generation and analysis.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">mutlu2023accelerating</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 60th Design Automation Conference (DAC)}</span><span class="p">,</span>
  <span class="na">series</span> <span class="p">=</span> <span class="s">{{DAC} '23}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Accelerating Genome Analysis via Algorithm-Architecture Co-Design}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Mutlu, Onur and Firtina, Can}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{arXiv preprint arXiv:2305.00492}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.48550/arXiv.2305.00492}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jul</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">MMDCS</abbr></div> <div id="Yavits2023" class="col-sm-8"> <div class="title">Will computing in memory become a new dawn of associative processors?</div> <div class="author"> <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>IEEE J. Emerg. Sel. Topics Circuits Syst.</em>, Jul 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://www.sciencedirect.com/science/article/pii/S2773064623000105" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://pdf.sciencedirectassets.com/781447/1-s2.0-S2773064622X00054/1-s2.0-S2773064623000105/main.pdf?X-Amz-Security-Token=IQoJb3JpZ2luX2VjEN7%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLWVhc3QtMSJHMEUCIQCK0yjv6dHgjtKcrpc%2Fa27U4MvadYsp6LVCgElF81GXtgIgKCLDvI1055z6N9cx%2F71s9YmBg2i%2FRYqsNXSlCXxRP08quwUIxv%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAFGgwwNTkwMDM1NDY4NjUiDGFvyUzQWQYNeyohpyqPBYBJLVTEd6HZ%2FfgLtsieP6Fhp%2BuR6yLxt0eyDAxAtJvK8NWwW1u2l%2FPGkxF%2BoE8ylPk1KfFXp4gGQyFZdhVPS4f9avWWq%2BCGUlbYK7lQ8A8411tCxpZZfOfG6a898XNUoKYTewRXt3NsLIAgoZczjWIUQhgVx76uWH3OpFhdxvUXtR486DTZG6VjU80SbuaXigmV6RHxG2voL0mmIP2bnl09843Svpq%2Balx7slOd0%2BXuutzkszSqSUz3C%2FrcK5nDxmUlhoPtFmoIIGtc4GufPqgTyrzZ55ostbsQFKxpsbp4sJ5i%2FCbxK4gb%2BKDaMbH6j%2BdDQ%2B6y0%2FaHWbSqTDzxjtXH6oXzkBAqW4zTKyzr%2BJ5EHfff1YjPyFF6pZzDBCqEfjKtp0sR2gRN2evmuPFXcHqpGzQULt8OGHXBfkmDDywIWfmHSNJdAEmcYvMwqUbz671grrbqaKQ%2Bdi1bubBtuidnQpxasBG0JaZi%2Ff%2BLQGRGIT2rGAzZTpvPwlLcydd7gm2KX7sPh7b0Ve0R7o%2FUrhRLGwbwowhC4XRQ9B90kDE3K2hcO09lTzijohk5K0iar0WKazKy%2FTZaJhSIYD108tsS5ebbC5f8aCHDejNIlixk2C22kbPfpOqQ40RD8YUs9pIf6mLit1SOPvOBJb35ZRZnKk3bSYf4sIvKn64PNCZO4PFNtwywEeAxKTtj%2Fl216pptF%2Bubwt%2BVIQMIwzNaRKYVHPTxCIT4Sm3H%2BrwNDaGunNKx2E0DF5vmp3bWw4TWIdx0iGSa%2Bss6JcATQFloez2%2B5boE2uD0LgoII2GC6oOGKhWqoWnWVjM4BN7iOdFGX1FH6IBITQ%2BwWI5d%2B%2BcbXypmk7zGYdRQNvwP%2FdSHeDowwL%2BLvAY6sQHP4S4XHRHGWK2mWLZXsuTunkaJ7Z3Bc24o57N%2Fy67OFP%2BXkcHg9zJ46UF%2FJBsmK0YZjY91PzIMP3vg1zOMJR%2FsTu51X%2BYTFn2fIUzEv3yJ4PtVo%2FUJWepZRalTTb%2FoBhekLBVUvoNmSNB5p0beSN7OGKa0rISvMRf7Dq7iFz%2FIAcdzG%2FIstdT2VQ2aSajEmSZeExcOIfHSKdr6TF58%2FPj06NQXlKfoawD2ZsQDgAVoljE%3D&amp;X-Amz-Algorithm=AWS4-HMAC-SHA256&amp;X-Amz-Date=20250111T220929Z&amp;X-Amz-SignedHeaders=host&amp;X-Amz-Expires=300&amp;X-Amz-Credential=ASIAQ3PHCVTY2IKND2KV%2F20250111%2Fus-east-1%2Fs3%2Faws4_request&amp;X-Amz-Signature=389e84204a167f3998c3737ae9d3e83bacf5cab50152995ed44194bd28addc47&amp;hash=13012dfdaea5b640d24f77f2b31e8f9dfe2decbe8405f2390bed6d8590bd88a1&amp;host=68042c943591013ac2b2430a89b270f6af2c76d8dfd086a07176afe7c76c2c61&amp;pii=S2773064623000105&amp;tid=spdf-95baa675-2a5e-4cd7-a24b-523d8c4f0bcc&amp;sid=ee3152a98dde27485d8941f0d1012cbba0eegxrqb&amp;type=client&amp;tsoh=d3d3LnNjaWVuY2VkaXJlY3QuY29t&amp;ua=16175855525e520551&amp;rr=90083b40ea027242&amp;cc=tr" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1016/j.memori.2023.100033" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1016/j.memori.2023.100033" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Computer architecture faces an enormous challenge in recent years: while the demand for performance is constantly growing, the performance improvement of general-purpose CPU has almost stalled. Among the reasons are memory and power walls, due to which data transfer increasingly dominates computing. By significantly reducing data transfer, data-centric (or in-memory) computing promises to alleviate the memory and power walls. Associative processor is a non von Neumann computer invented in the 1960s but effectively cast aside until recently. It computes using associative memory in a perfect induction like fashion, using associative memory cells for both data storage and processing. Associative processor can be implemented using conventional CMOS as well as emerging memories. We show that associative processor can outperform state-of-the-art computing platforms by up to almost two orders of magnitude in a variety of data-intensive workloads.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">Yavits2023</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Will computing in memory become a new dawn of associative processors?}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{{IEEE} J. Emerg. Sel. Topics Circuits Syst.}</span><span class="p">,</span>
  <span class="na">issn</span> <span class="p">=</span> <span class="s">{2773-0646}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{100033}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{4}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1016/j.memori.2023.100033}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jul</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">AACBB</abbr></div> <div id="abecasis2023" class="col-sm-8"> <div class="title">GAPiM: a hardware acceleration of Genome Analysis pipeline using Processing in Memory</div> <div class="author"> <a href="https://www.linkedin.com/in/naomie-abecassis-361661273/" rel="external nofollow noopener" target="_blank">Naomie Abecassis</a>, <a href="https://ee.ethz.ch/the-department/people-a-z/person-detail.MjQxMjgy.TGlzdC8zMjc5LC0xNjUwNTg5ODIw.html" rel="external nofollow noopener" target="_blank">Juan Gómez-Luna</a>, <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a>, <a href="https://webee.technion.ac.il/~ran/" rel="external nofollow noopener" target="_blank">Ran Ginosar</a>, <a href="https://www.linkedin.com/in/a-moisson/" rel="external nofollow noopener" target="_blank">Aphélie Moisson-Franckhauser</a>, and <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>In Proceedings of the 5th Workshop on Accelerator Architecture in Computational Biology and Bioinformatics (AACBB)</em>, Jun 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://www.biorxiv.org/content/10.1101/2023.07.26.550623v1" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://github.com/naomieab/UPMEM_HAPLOTYPECALLER" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1101/2023.07.26.550623v1" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1101/2023.07.26.550623v1" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Variant calling is a fundamental stage in genome analysis that identifies mutations (variations) in a sequenced genome relative to a known reference genome. Pair-HMM is a key part of the variant calling algorithm and its most compute-intensive part. In recent years, Processing-in-Memory (PiM) solutions, which consist of placing compute capabilities near/inside memory, have been proposed to speed up the genome analysis pipeline. We implement the Pair-HMM algorithm on a commercial PiM platform developed by UPMEM. We modify the Pair-HMM algorithm to make it more suitable for PiM execution with acceptable loss of accuracy. We evaluate our implementation on single chromosomes and whole genome sequencing datasets, demonstrating up to 2x speedup compared to existing CPU accelerations and up to 3x speedup compared to FPGA accelerations.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">abecasis2023</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 5th Workshop on Accelerator Architecture in Computational Biology and Bioinformatics (AACBB)}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{{GAPiM}: a hardware acceleration of Genome Analysis pipeline using Processing in Memory}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1101/2023.07.26.550623v1}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Abecassis, Naomie and G{\'o}mez-Luna, Juan and Mutlu, Onur and Ginosar, Ran and Moisson-Franckhauser, Aph{\'e}lie and Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jun</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">ISPASS</abbr></div> <div id="gomez2023transpimlib" class="col-sm-8"> <div class="title">TransPimLib: A Library for Efficient Transcendental Functions on Processing-in-Memory Systems</div> <div class="author"> Maurus Item, <a href="https://ee.ethz.ch/the-department/people-a-z/person-detail.MjQxMjgy.TGlzdC8zMjc5LC0xNjUwNTg5ODIw.html" rel="external nofollow noopener" target="_blank">Juan Gómez-Luna</a>, Yuxin Guo, Geraldo F Oliveira, Mohammad Sadrosadati, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em>In Proceedings of the 24th International Symposium on Performance Analysis of Systems and Software (ISPASS)</em>, Apr 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.org/10.48550/arXiv.2304.01951" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://arxiv.org/pdf/2304.01951" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.48550/arXiv.2304.01951" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.48550/arXiv.2304.01951" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Processing-in-memory (PIM) promises to alleviate the data movement bottleneck in modern computing systems. However, current real-world PIM systems have the inherent disadvantage that their hardware is more constrained than in conventional processors (CPU, GPU), due to the difficulty and cost of building processing elements near or inside the memory. As a result, general-purpose PIM architectures support fairly limited instruction sets and struggle to execute complex operations such as transcendental functions and other hard-to-calculate operations (e.g., square root). These operations are particularly important for some modern workloads, e.g., activation functions in machine learning applications. In order to provide support for transcendental (and other hard-to-calculate) functions in general-purpose PIM systems, we present TransPimLib, a library that provides CORDIC-based and LUT-based methods for trigonometric functions, hyperbolic functions, exponentiation, logarithm, square root, etc. We develop an implementation of TransPimLib for the UPMEM PIM architecture and perform a thorough evaluation of TransPimLib’s methods in terms of performance and accuracy, using microbenchmarks and three full workloads (Blackscholes, Sigmoid, Softmax). We open-source all our code and datasets at https://github.com/CMU-SAFARI/transpimlib.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">gomez2023transpimlib</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 24th International Symposium on Performance Analysis of Systems and Software (ISPASS)}</span><span class="p">,</span>
  <span class="na">series</span> <span class="p">=</span> <span class="s">{{ISPASS} '23}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{TransPimLib: A Library for Efficient Transcendental Functions on Processing-in-Memory Systems}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Item, Maurus and G{\'o}mez-Luna, Juan and Guo, Yuxin and Oliveira, Geraldo F and Sadrosadati, Mohammad and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{arXiv preprint arXiv:2304.01951}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.48550/arXiv.2304.01951}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">apr</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">ISPASS</abbr></div> <div id="gomez2023evaluating" class="col-sm-8"> <div class="title">Evaluating Machine Learning Workloads on Memory-Centric Computing Systems</div> <div class="author"> <a href="https://ee.ethz.ch/the-department/people-a-z/person-detail.MjQxMjgy.TGlzdC8zMjc5LC0xNjUwNTg5ODIw.html" rel="external nofollow noopener" target="_blank">Juan Gómez-Luna</a>, Yuxin Guo, Sylvan Brocard, Julien Legriel, Remy Cimadomo, Geraldo F Oliveira, <a href="https://scholar.google.ch/citations?hl=en&amp;user=tb097i4AAAAJ" rel="external nofollow noopener" target="_blank">Gagandeep Singh</a>, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em>In Proceedings of the 24th International Symposium on Performance Analysis of Systems and Software (ISPASS)</em>, Apr 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://arxiv.org/abs/2207.07886" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://arxiv.org/pdf/2207.07886" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.48550/arXiv.2207.07886" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.48550/arXiv.2207.07886" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Training machine learning (ML) algorithms is a computationally intensive process, which is frequently memory-bound due to repeatedly accessing large training datasets. As a result, processor-centric systems (e.g., CPU, GPU) suffer from costly data movement between memory units and processing units, which consumes large amounts of energy and execution cycles. Memory-centric computing systems, i.e., with processing-in-memory (PIM) capabilities, can alleviate this data movement bottleneck. Our goal is to understand the potential of modern general-purpose PIM architectures to accelerate ML training. To do so, we (1) implement several representative classic ML algorithms (namely, linear regression, logistic regression, decision tree, K-Means clustering) on a real-world general-purpose PIM architecture, (2) rigorously evaluate and characterize them in terms of accuracy, performance and scaling, and (3) compare to their counterpart implementations on CPU and GPU. Our evaluation on a real memory-centric computing system with more than 2500 PIM cores shows that general-purpose PIM architectures can greatly accelerate memory-bound ML workloads, when the necessary operations and datatypes are natively supported by PIM hardware. For example, our PIM implementation of decision tree is 27x faster than a state-of-the-art CPU version on an 8-core Intel Xeon, and 1.34x faster than a state-of-the-art GPU version on an NVIDIA A100. Our K-Means clustering on PIM is 2.8x and 3.2x than state-of-the-art CPU and GPU versions, respectively. To our knowledge, our work is the first one to evaluate ML training on a real-world PIM architecture. We conclude with key observations, takeaways, and recommendations that can inspire users of ML workloads, programmers of PIM architectures, and hardware designers &amp; architects of future memory-centric computing systems.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">gomez2023evaluating</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 24th International Symposium on Performance Analysis of Systems and Software (ISPASS)}</span><span class="p">,</span>
  <span class="na">series</span> <span class="p">=</span> <span class="s">{{ISPASS} '23}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Evaluating Machine Learning Workloads on Memory-Centric Computing Systems}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{G{\'o}mez-Luna, Juan and Guo, Yuxin and Brocard, Sylvan and Legriel, Julien and Cimadomo, Remy and Oliveira, Geraldo F and Singh, Gagandeep and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{arXiv preprint arXiv:2207.07886}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.48550/arXiv.2207.07886}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">apr</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">Bioinformatics</abbr></div> <div id="Lindegger2023" class="col-sm-8"> <div class="title">Scrooge: A Fast and Memory-Frugal Genomic Sequence Aligner for CPUs, GPUs, and ASICs.</div> <div class="author"> Joël Lindegger, <a href="https://damlasenolcali.github.io" rel="external nofollow noopener" target="_blank">Damla Senol Cali</a>, <a href="https://mealser.github.io" rel="external nofollow noopener" target="_blank">Mohammed Alser</a>, <a href="https://ee.ethz.ch/the-department/people-a-z/person-detail.MjQxMjgy.TGlzdC8zMjc5LC0xNjUwNTg5ODIw.html" rel="external nofollow noopener" target="_blank">Juan Gómez-Luna</a>, <a href="https://scholar.google.ch/citations?user=lMuBESUAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Nika Mansouri Ghiasi</a>, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em>Bioinformatics</em>, Mar 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.org/10.1093/bioinformatics/btad151" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://watermark.silverchair.com/btad151.pdf?token=AQECAHi208BE49Ooan9kkhW_Ercy7Dm3ZL_9Cf3qfKAc485ysgAAA3AwggNsBgkqhkiG9w0BBwagggNdMIIDWQIBADCCA1IGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMezfmAZFf_GzHIoZBAgEQgIIDIwkQjgDECIpwPHw2Yy15-8CStHBgNPmVPYeci8NJTzeKow-HC1C_2iEmlU8xEqsYbpMrRkD1fjoD_Pg_btCIUFkHY8qYmCLSLflRKAOQSIVJO_92rcRlcTmEQ5y_A1OkAJqs0s0Gqm9il0m8wd2VYg7Zy80AH9xB26lYU-65IvNutWdGmZYAkk6d0KipyeI8jEZtW3QZF22VwMEAqyAmt550zGy0K6HAFF-FmJXRw5VNZhvrwGVFhN4o-QnxIMJu3SKKhsDzGPZGUMlvauSn502dckX034_SO0_XGg9ZzJpNldjphmOjkMrzz-sIH77nBM9HOMUo3vDFAhwTI7maVjQVc-BBwKkFtJJsXSMSGLsgFWXJPDXDcPcdEM8vbyFy84Ur_QZI6BdKTUf7gAd0r8wI7_frHrI2vPJT1Je6DHUtvwcdX-6MuKQI7Rk-48sqR1KCHapH7DncF90ChhAplEgIGb2xLP_lYonFyZKG2WIxH2xJ9zOmJ2Nfx7TarxPNza5-aR26kQEJpNOSEgwuOrmkzhaGt89zbHao13C1duX2BefXuF9MYaPlt-gEj0K3_jL7PFEcGN5Q7BZotsIjEqNTBFyK_zu14-umSYi_Y3EubWKaomLDa4kONvoxdwldcNvkt0kaJch6XR_c-qaS1touo8uqdVZhe3_E5ft1Cb2be2kYqErBuHw3EureNoWc55VxLYyxtEZpxQq1tw1Y8BM8mFiPJIDw0xCz6ClF5oZ86R8RD8nI0aT2mWyXh6LDIFYWT2TyWYRLy6vuNXugyEEJhWYQ-5jSsy-JYhrQ0Ln4t1UEPH0tZMjUbuSazHuLbLV3-3UAH15UVkKA8zAD4JFiOBasl2_rEvf-_QNMv944n4uwQNa7cYktSVxsc7U_0i4854thW32GqZ_Isptc9xlLaNc9tIO1bUgo0_mlsCFHRgWrgyfzvVO2Hw-20R-c4TCfOz2janTxPDIj4zdwcFWJQdodpm6TnJAABDHk0J2QpUZikc8kl6QjirW6Ptwg7TidVKvMtIPbQROQ4R7LLtlSRAS33a0w8e1SyjG5HYQ4m7Nw" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1093/bioinformatics/btad151" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1093/bioinformatics/btad151" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Pairwise sequence alignment is a very time-consuming step in common bioinformatics pipelines. Speeding up this step requires heuristics, efficient implementations, and/or hardware acceleration. A promising candidate for all of the above is the recently proposed GenASM algorithm. We identify and address three inefficiencies in the GenASM algorithm: it has a high amount of data movement, a large memory footprint, and does some unnecessary work. We propose Scrooge, a fast and memory-frugal genomic sequence aligner. Scrooge includes three novel algorithmic improvements which reduce the data movement, memory footprint, and the number of operations in the GenASM algorithm. We provide efficient open-source implementations of the Scrooge algorithm for CPUs and GPUs, which demonstrate the significant benefits of our algorithmic improvements. For long reads, the CPU version of Scrooge achieves a 20.1x, 1.7x, and 2.1x speedup over KSW2, Edlib, and a CPU implementation of GenASM, respectively. The GPU version of Scrooge achieves a 4.0x 80.4x, 6.8x, 12.6x and 5.9x speedup over the CPU version of Scrooge, KSW2, Edlib, Darwin-GPU, and a GPU implementation of GenASM, respectively. We estimate an ASIC implementation of Scrooge to use 3.6x less chip area and 2.1x less power than a GenASM ASIC while maintaining the same throughput. Further, we systematically analyze the throughput and accuracy behavior of GenASM and Scrooge under various configurations. As the best configuration of Scrooge depends on the computing platform, we make several observations that can help guide future implementations of Scrooge. https://github.com/CMU-SAFARI/Scrooge.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">Lindegger2023</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lindegger, Joël and Cali, Damla Senol and Alser, Mohammed and Gómez-Luna, Juan and Ghiasi, Nika Mansouri and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Bioinformatics}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Scrooge: A Fast and Memory-Frugal Genomic Sequence Aligner for {CPUs}, {GPUs}, and {ASICs}.}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">issn</span> <span class="p">=</span> <span class="s">{1367-4811}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">citation-subset</span> <span class="p">=</span> <span class="s">{IM}</span><span class="p">,</span>
  <span class="na">country</span> <span class="p">=</span> <span class="s">{England}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1093/bioinformatics/btad151}</span><span class="p">,</span>
  <span class="na">issn-linking</span> <span class="p">=</span> <span class="s">{1367-4803}</span><span class="p">,</span>
  <span class="na">nlm-id</span> <span class="p">=</span> <span class="s">{9808944}</span><span class="p">,</span>
  <span class="na">owner</span> <span class="p">=</span> <span class="s">{NLM}</span><span class="p">,</span>
  <span class="na">pii</span> <span class="p">=</span> <span class="s">{btad151}</span><span class="p">,</span>
  <span class="na">pmid</span> <span class="p">=</span> <span class="s">{36961334}</span><span class="p">,</span>
  <span class="na">pubmodel</span> <span class="p">=</span> <span class="s">{Print-Electronic}</span><span class="p">,</span>
  <span class="na">pubstate</span> <span class="p">=</span> <span class="s">{aheadofprint}</span><span class="p">,</span>
  <span class="na">revised</span> <span class="p">=</span> <span class="s">{2023-03-24}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">Bioinformatics</abbr></div> <div id="Diab2023" class="col-sm-8"> <div class="title">A framework for high-throughput sequence alignment using real processing-in-memory systems.</div> <div class="author"> Safaa Diab, Amir Nassereldine, <a href="https://mealser.github.io" rel="external nofollow noopener" target="_blank">Mohammed Alser</a>, <a href="https://ee.ethz.ch/the-department/people-a-z/person-detail.MjQxMjgy.TGlzdC8zMjc5LC0xNjUwNTg5ODIw.html" rel="external nofollow noopener" target="_blank">Juan Gómez Luna</a>, <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a>, and Izzat El Hajj</div> <div class="periodical"> <em>Bioinformatics</em>, Mar 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.org/10.1093/bioinformatics/btad155" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://watermark.silverchair.com/btad155.pdf?token=AQECAHi208BE49Ooan9kkhW_Ercy7Dm3ZL_9Cf3qfKAc485ysgAAA3AwggNsBgkqhkiG9w0BBwagggNdMIIDWQIBADCCA1IGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMCw1JFMWz-dB_twiSAgEQgIIDI3s5K2eczRu9Q0CgCbOUlBqjlpz9IfJzFZvVfAFq2AnkyFZvAy2isZtHwVsRLckNnMyFbjY-2DSXU7Z0fyN2shCHXbWsqvnMQCKdqArvBr_7jp7KEINJQGhS9FU4hH8Vqeo59Ic-4VsbA0wi0QlGZpClzFPm24WjJTUblRkLagx_sd0dN1UF1Akju-svLKkEkkJIItSpvgh2wQsjy2NgBJhw1qZ3l2Ue4aXoINqULNq-ZSPeJ9lgeb9Y6minDNGPdQZ8nIVl6ICwonk5OgLProiZAHO6BVkhrAvkj2eU-wsx6SF5ZVLomsuR_wbdB2gz9repgx2zt61hsQhCOGeK9U8OHbeOnVQz1vg1WawB9HSDlYwpUmhNQXmzWGzsB81W_kcDCB_tW2oBLwe1q3e1rc9_L_t5q9l4Qd-uhk7o1vmQTcXDIcdCAGrLzrTnoGTB2G1YZpaUBVrbKrVhs9YBvkUKC7OkpZPjBib1y5iOU1LD5mmVwWlqf5rqd25PX7PYUWt4HRjwEOMeApf_XIFLY6LmijMbhsATxFcdQqoZlKYU6J-4RxrWNIxzOugGOyGbDK_7I_5RAxlTUVdM7RFtzwC2R01_OeShzN4AXcpNPlr75GBg-w4ZH7qs2PYa45SzwymBDnleL3bWP-8nyQ_3ZpOylSHJIjSksXsTm93ngqN-aEPQ9CMz1FWRbuZa81MCw3QYoGL4On1z8XXGgemieh7ZzYt8qzx0GIuXboqqxqXctqLzIU2bIKdce3Q1BQ7Vg9emJMsoptrdulFRV-5plH_GhNayIvRyuqJ19K5nrKNMc8hSUtkNJrU3Xgl_N-W6qTp-1RyIOJXQGIWZmNBLppeLk0_k_Af0Zet5iG6upb8gSwxmDOTNfs4H92nhrvuQtr2FMeb_zHkdG6346YofMn-nEYWo5HOLpQA8rdHSvpySFouk2vZnlmaQJMj4vZ2v2UBAR3u7wISJymEJSjVamEZq_ITkk2GS6i6fvSymG9OvxAwEjml_fAXiY4QqMiWI2E4Ruu5FdnPXSRNokiFnL3Bs1O5DWUh_3Iwfj7hqEKwOCRan" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1093/bioinformatics/btad155" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1093/bioinformatics/btad155" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Sequence alignment is a memory bound computation whose performance in modern systems is limited by the memory bandwidth bottleneck. Processing-in-memory (PIM) architectures alleviate this bottleneck by providing the memory with computing competencies. We propose Alignment-in-Memory (AIM), a framework for high-throughput sequence alignment using PIM, and evaluate it on UPMEM, the first publicly available general-purpose programmable PIM system. Our evaluation shows that a real PIM system can substantially outperform server-grade multi-threaded CPU systems running at full-scale when performing sequence alignment for a variety of algorithms, read lengths, and edit distance thresholds. We hope that our findings inspire more work on creating and accelerating bioinformatics algorithms for such real PIM systems. Our code is available at https://github.com/safaad/aim.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">Diab2023</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Diab, Safaa and Nassereldine, Amir and Alser, Mohammed and Gómez Luna, Juan and Mutlu, Onur and El Hajj, Izzat}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Bioinformatics}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{A framework for high-throughput sequence alignment using real processing-in-memory systems.}</span><span class="p">,</span>
  <span class="na">issn</span> <span class="p">=</span> <span class="s">{1367-4811}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{39}</span><span class="p">,</span>
  <span class="na">citation-subset</span> <span class="p">=</span> <span class="s">{IM}</span><span class="p">,</span>
  <span class="na">completed</span> <span class="p">=</span> <span class="s">{2023-05-08}</span><span class="p">,</span>
  <span class="na">country</span> <span class="p">=</span> <span class="s">{England}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1093/bioinformatics/btad155}</span><span class="p">,</span>
  <span class="na">issn-linking</span> <span class="p">=</span> <span class="s">{1367-4803}</span><span class="p">,</span>
  <span class="na">issue</span> <span class="p">=</span> <span class="s">{5}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Sequence Alignment; Algorithms; Software; Computational Biology; Sequence Analysis, DNA; High-Throughput Nucleotide Sequencing}</span><span class="p">,</span>
  <span class="na">nlm-id</span> <span class="p">=</span> <span class="s">{9808944}</span><span class="p">,</span>
  <span class="na">owner</span> <span class="p">=</span> <span class="s">{NLM}</span><span class="p">,</span>
  <span class="na">pii</span> <span class="p">=</span> <span class="s">{btad155}</span><span class="p">,</span>
  <span class="na">pmc</span> <span class="p">=</span> <span class="s">{PMC10159653}</span><span class="p">,</span>
  <span class="na">pmid</span> <span class="p">=</span> <span class="s">{36971586}</span><span class="p">,</span>
  <span class="na">pubmodel</span> <span class="p">=</span> <span class="s">{Print}</span><span class="p">,</span>
  <span class="na">pubstate</span> <span class="p">=</span> <span class="s">{ppublish}</span><span class="p">,</span>
  <span class="na">revised</span> <span class="p">=</span> <span class="s">{2023-05-09}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">JETCAS</abbr></div> <div id="Garzon2023" class="col-sm-8"> <div class="title">AM4: MRAM Crossbar Based CAM/TCAM/ACAM/AP for In-Memory Computing</div> <div class="author"> Esteban Garzón, Marco Lanuzza, Adam Teman, and <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>IEEE J. Emerg. Sel. Topics Circuits Syst.</em>, Mar 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://ieeexplore.ieee.org/abstract/document/10040683" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&amp;arnumber=10040683" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1109/JETCAS.2023.3243222" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1109/JETCAS.2023.3243222" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>In-memory computing seeks to minimize data movement and alleviate the memory wall by computing in-situ, in the same place that the data is located. One of the key emerging technologies that promises to enable such computing-in-memory is spin-transfer torque magnetic tunnel junction (STT-MTJ). This paper proposes AM4, a combined STT-MTJ-based Content Addressable Memory (CAM), Ternary CAM (TCAM), approximate matching (similarity search) CAM (ACAM), and in-memory Associative Processor (AP) design, inspired by the recently announced Samsung MRAM crossbar. We demonstrate and evaluate the performance and energy-efficiency of the AM4-based AP using a variety of data intensive workloads. We show that an AM4-based AP outperforms state-of-the-art solutions both in performance (with the average speedup of about 10 x) and energy-efficiency (by about 60 x on average).</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">Garzon2023</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Garz{\'{o}}n, Esteban and Lanuzza, Marco and Teman, Adam and Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{{IEEE} J. Emerg. Sel. Topics Circuits Syst.}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{AM4: {MRAM} Crossbar Based {CAM/TCAM/ACAM/AP} for In-Memory Computing}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{1}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{408--421}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{13}</span><span class="p">,</span>
  <span class="na">bibsource</span> <span class="p">=</span> <span class="s">{dblp computer science bibliography, https://dblp.org}</span><span class="p">,</span>
  <span class="na">biburl</span> <span class="p">=</span> <span class="s">{https://dblp.org/rec/journals/esticas/GarzonLTY23.bib}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1109/JETCAS.2023.3243222}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">NARGAB</abbr></div> <div id="firtina_blend_2023" class="col-sm-8"> <div class="title">BLEND: a fast, memory-efficient and accurate mechanism to find fuzzy seed matches in genome analysis</div> <div class="author"> <a href="https://cfirtina.com" rel="external nofollow noopener" target="_blank">Can Firtina</a>, Jisung Park, <a href="https://mealser.github.io" rel="external nofollow noopener" target="_blank">Mohammed Alser</a>, <a href="https://scholar.google.com/citations?user=-pOH1s4AAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Jeremie S Kim</a>, Damla Senol Cali, Taha Shahroodi, Nika Mansouri Ghiasi, <a href="https://scholar.google.ch/citations?hl=en&amp;user=tb097i4AAAAJ" rel="external nofollow noopener" target="_blank">Gagandeep Singh</a>, Konstantinos Kanellopoulos, <a href="https://www.alkanlab.org/team/can-alkan/" rel="external nofollow noopener" target="_blank">Can Alkan</a>, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em>NAR Genomics and Bioinformatics</em>, Mar 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://arxiv.org/pdf/2112.08687" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://github.com/CMU-SAFARI/BLEND" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1093/nargab/lqad004" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1093/nargab/lqad004" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Generating the hash values of short subsequences, called seeds, enables quickly identifying similarities between genomic sequences by matching seeds with a single lookup of their hash values. However, these hash values can be used only for finding exact-matching seeds as the conventional hashing methods assign distinct hash values for different seeds, including highly similar seeds. Finding only exact-matching seeds causes either (i) increasing the use of the costly sequence alignment or (ii) limited sensitivity. We introduce BLEND, the first efficient and accurate mechanism that can identify both exact-matching and highly similar seeds with a single lookup of their hash values, called fuzzy seed matches. BLEND (i) utilizes a technique called SimHash, that can generate the same hash value for similar sets, and (ii) provides the proper mechanisms for using seeds as sets with the SimHash technique to find fuzzy seed matches efficiently. We show the benefits of BLEND when used in read overlapping and read mapping. For read overlapping, BLEND is faster by 2.4x–83.9x (on average 19.3x), has a lower memory footprint by 0.9x–14.1x (on average 3.8x), and finds higher quality overlaps leading to accurate de novo assemblies than the state-of-the-art tool, minimap2. For read mapping, BLEND is faster by 0.8x–4.1x (on average 1.7x) than minimap2. Source code is available at https://github.com/CMU-SAFARI/BLEND.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">firtina_blend_2023</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{{BLEND}: a fast, memory-efficient and accurate mechanism to find fuzzy seed matches in genome analysis}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{5}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1093/nargab/lqad004}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{1}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{NAR Genomics and Bioinformatics}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Firtina, Can and Park, Jisung and Alser, Mohammed and Kim, Jeremie S and Cali, Damla Senol and Shahroodi, Taha and Ghiasi, Nika Mansouri and Singh, Gagandeep and Kanellopoulos, Konstantinos and Alkan, Can and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{lqad004}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">Chips</abbr></div> <div id="Garzon2023a" class="col-sm-8"> <div class="title">Approximate Content-Addressable Memories: A Review</div> <div class="author"> Esteban Garzón, <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a>, Adam Teman, and Marco Lanuzza</div> <div class="periodical"> <em>Chips</em>, Mar 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://www.mdpi.com/2674-0729/2/2/5" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.3390/chips2020005" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.3390/chips2020005" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Content-addressable memory (CAM) has been part of the memory market for more than five decades. CAM can carry out a single clock cycle lookup based on the content rather than an address. Thanks to this attractive feature, CAM is utilized in memory systems where a high-speed content lookup technique is required. However, typical CAM applications only support exact matching, as opposed to approximate matching, where a certain Hamming distance (several mismatching characters between a query pattern and the dataset stored in CAM) needs to be tolerated. Recent interest in approximate search has led to the development of new CAM-based alternatives, accelerating the processing of large data workloads in the realm of big data, genomics, and other data-intensive applications. In this review, we provide an overview of approximate CAM and describe its current and potential applications that would benefit from approximate search computing.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">Garzon2023a</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Garzón, Esteban and Yavits, Leonid and Teman, Adam and Lanuzza, Marco}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Chips}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Approximate Content-Addressable Memories: A Review}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">issn</span> <span class="p">=</span> <span class="s">{2674-0729}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{2}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{70--82}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{2}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.3390/chips2020005}</span><span class="p">,</span>
  <span class="na">owner</span> <span class="p">=</span> <span class="s">{calkan}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://www.mdpi.com/2674-0729/2/2/5}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> </ol> </div> <h1>Posters</h1> <div class="publications"> <h2 class="year">2024</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">GI</abbr></div> <div id="bingol2024poster" class="col-sm-8"> <div class="title">Hardware/software co-design for sequence-to-pangenome mapping</div> <div class="author"> <a href="http://www.alkanlab.org/team/z%C3%BClal-bing%C3%B6l/" rel="external nofollow noopener" target="_blank">Zülal Bingöl</a>, Ccan Fırtına, Konstantina Koliogeorgi, <a href="http://www.alkanlab.org/team/ricardo-roman-brenes/" rel="external nofollow noopener" target="_blank">Ricardo Roman-Brenes</a>, <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a>, and <a href="https://www.alkanlab.org/team/can-alkan/" rel="external nofollow noopener" target="_blank">Can Alkan</a> </div> <div class="periodical"> <em>In Genome Informatics Conference 2024</em>, Nov 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://coursesandconferences.wellcomeconnectingscience.org/event/genome-informatics-20241113/" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>The reference genome is the representative genome of a species. Most genomics analyses begin with or involve a comparison with the reference genome of the same species, which serves as a template for the newly sequenced sample of an individual. However, using a single reference genome creates reference bias since it does not contain information about all individuals of a species to represent the genomic diversity. The rapid development of sequencing technologies has led to an increasing number of samples being studied. With the growing number of genetic materials sampled, genomics pursues utilizing this data to express all samples and capture vast genetic diversity. So, to alleviate these problems and enable diverse variance studies (e.g., bacterial pathogenicity, antibacterial resistance, virulence, human leukocyte antigens), pangenome structures are proposed. As the efforts to collect more data to improve the diversity representation on the pangenome are growing, the computational methods to assist this field become more vital to meet the high demand. Therefore, there is a pressing need for acceleration methods tuned for mapping sequences to pangenomes. To address this problem, in this work, we propose a hardware/software co-design framework for optimizing the pangenome graphs for compact storage and developing efficient algorithms that eventually lead to fast and accurate sequence-to-pangenome mapping. Our framework combines several steps (e.g., indexing, seeding, and alignment), and the accelerator includes optimized designs based on each step’s requirements.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex">  <span class="c">booktitle = {Genome Informatics Conference 2024},</span>
  <span class="c">series = {{GI} '24},</span>
  <span class="c">title = {Hardware/software co-design for sequence-to-pangenome mapping},</span>
  <span class="c">author = {Bingöl, Zülal and Fırtına, Ccan and Koliogeorgi, Konstantina and Roman-Brenes, Ricardo and Mutlu, Onur and Alkan, Can},</span>
  <span class="c">journal = {Poster presented},</span>
  <span class="c">url = {https://coursesandconferences.wellcomeconnectingscience.org/event/genome-informatics-20241113/},</span>
  <span class="c">year = {2024},</span>
  <span class="c">month = nov,</span>
<span class="c">}</span></code></pre></figure> </div> </div> </div> </li></ol> <h2 class="year">2023</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">RECOMB</abbr></div> <div id="ilgun2023poster" class="col-sm-8"> <div class="title">Characterization of Alignment and Search Algorithms for Short Read, Long Read, and Graph Mappers</div> <div class="author"> <a href="http://www.alkanlab.org/team/ecem-ilg%C3%BCn/" rel="external nofollow noopener" target="_blank">Ecem İlgün</a>, <a href="http://www.alkanlab.org/team/%C3%B6mer-yavuz-%C3%B6zt%C3%BCrk/" rel="external nofollow noopener" target="_blank">Ömer Yavuz Öztürk</a>, <a href="http://www.alkanlab.org/team/klea-zambaku/" rel="external nofollow noopener" target="_blank">Klea Zambaku</a>, Juan Gómez Luna, <a href="https://mealser.github.io" rel="external nofollow noopener" target="_blank">Mohammed Alser</a>, <a href="http://www.alkanlab.org/team/ricardo-roman-brenes/" rel="external nofollow noopener" target="_blank">Ricardo Román-Brenes</a>, The BioPIM Project, and <a href="https://www.alkanlab.org/team/can-alkan/" rel="external nofollow noopener" target="_blank">Can Alkan</a> </div> <div class="periodical"> <em>In RECOMB 2023</em>, Apr 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://docs.google.com/document/d/1OK2oSsDjQYuo20jcJXwf8H5H_PMF6t_MOWO9QZOkFPg/view" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>We recently started a project funded by the Horizon Europe program, BioPIM, which aims to accelerate various bioinformatics algorithms using processing-in-memory (PIM) technologies. PIM is a type of computer architecture that aims to solve the issue of data movement between the CPU and memory being a bottleneck in data-intensive applications. PIM accomplishes this by integrating computing units directly into the memory chip, which reduces latency and increases bandwidth by bringing the computing units closer to the memory. Since read mapping is a crucial step in almost all genome analysis studies, we first aimed to understand how to accelerate read mapping in this project. As the first step, we analyzed the computational workload of BWA-MEM and Bowtie2 for short reads, NGMLR, Minimap2 and LRA for long reads, and finally minigraph, vg, GraphAligner and GWFA for read-to-graph aligners. Here we present a first-pass workload analysis using the Intel vtune tool to identify the most time and memory-bandwidth consuming functions used by these algorithms. Our preliminary results show that the resource usage of these algorithms varied significantly depending on the type of data and the algorithm used. We have identified several potential functions that could potentially be improved by PIM. Furthermore, we discuss processing-in-memory architectures to accelerate alignment and search algorithms for resequencing experiments. Overall, our study provides insights into the resource requirements of different alignment and search algorithms for different types of sequencing data, which can guide the selection of the most appropriate algorithms for different resequencing experiments. Our findings can also inform the development of more efficient and accurate algorithms for processing sequencing data, which is critical for advancing our understanding of the genetic basis of complex diseases.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex">  <span class="c">booktitle = {RECOMB 2023},</span>
  <span class="c">series = {{RECOMB} '23},</span>
  <span class="c">title = {Characterization of Alignment and Search Algorithms for Short Read, Long Read, and Graph Mappers},</span>
  <span class="c">author = {İlgün, Ecem and Öztürk, Ömer Yavuz and Zambaku, Klea and Luna, Juan Gómez and Alser, Mohammed and Román-Brenes, Ricardo and Project, The BioPIM and Alkan, Can},</span>
  <span class="c">journal = {Poster presented},</span>
  <span class="c">url = {https://docs.google.com/document/d/1OK2oSsDjQYuo20jcJXwf8H5H_PMF6t_MOWO9QZOkFPg/view},</span>
  <span class="c">year = {2023},</span>
  <span class="c">month = apr,</span>
<span class="c">}</span></code></pre></figure> </div> </div> </div> </li></ol> </div> <h1>Related Publications</h1> <div class="publications"> <h2 class="year">2022</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">ISCA</abbr></div> <div id="hanhan_edam_2022" class="col-sm-8"> <div class="title">EDAM: Edit Distance Tolerant Approximate Matching Content Addressable Memory</div> <div class="author"> Robert Hanhan, Esteban Garzón, <a href="https://www.linkedin.com/in/zuher-jahshan-7a7199196/" rel="external nofollow noopener" target="_blank">Zuher Jahshan</a>, Adam Teman, Marco Lanuzza, and <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a> </div> <div class="periodical"> <em>In Proceedings of the 49th Annual International Symposium on Computer Architecture</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.org/10.1145/3470496.3527424" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1145/3470496.3527424" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1145/3470496.3527424" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>We propose a novel edit distance-tolerant content addressable memory (EDAM) for energy-efficient approximate search applications. Unlike state-of-the-art approximate search solutions that tolerate certain Hamming distance between the query pattern and the stored data, EDAM tolerates edit distance, which makes it especially efficient in applications such as text processing and genome analysis. EDAM was designed using a commercial 65 nm 1.2 V CMOS technology and evaluated through extensive Monte Carlo simulations, while considering different process corners. Simulation results show that EDAM can achieve robust approximate search operation with a wide range of edit distance threshold levels. EDAM is functionally evaluated as a pathogen DNA detection and classification accelerator. EDAM achieves up to 1.7x higher F1 score for high-quality DNA reads and up to 19.55x higher F1 score for DNA reads with 15% error rate, compared to state-of-the-art DNA classification tool Kraken2. Simulated at 667 MHz, EDAM provides 1, 214x average speedup over Kraken2. This makes EDAM suitable for hardware acceleration of genomic surveillance of outbreaks, such as the ongoing Covid-19 pandemic.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">hanhan_edam_2022</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">{New York, NY, USA}</span><span class="p">,</span>
  <span class="na">series</span> <span class="p">=</span> <span class="s">{{ISCA} '22}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{{EDAM}: {Edit} {Distance} {Tolerant} {Approximate} {Matching} {Content} {Addressable} {Memory}}</span><span class="p">,</span>
  <span class="na">isbn</span> <span class="p">=</span> <span class="s">{978-1-4503-8610-4}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://doi.org/10.1145/3470496.3527424}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1145/3470496.3527424}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 49th {Annual} {International} {Symposium} on {Computer} {Architecture}}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Association for Computing Machinery}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Hanhan, Robert and Garzón, Esteban and Jahshan, Zuher and Teman, Adam and Lanuzza, Marco and Yavits, Leonid}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2022}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{495--507}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li></ol> <h2 class="year">2021</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">VLSI Tech.</abbr></div> <div id="khaddam-aljameh_hermes_2021" class="col-sm-8"> <div class="title">HERMES Core – A 14nm CMOS and PCM-based In-Memory Compute Core using an array of 300ps/LSB Linearized CCO-based ADCs and local digital processing</div> <div class="author"> R. Khaddam-Aljameh, M. Stanisavljevic, J. Fornt Mas, G. Karunaratne, M. Braendli, F. Liu, A. Singh, S. M. Müller, U. Egger, A. Petropoulos, T. Antonakopoulos, K. Brew, S. Choi, I. Ok, F. L. Lie, N. Saulnier, V. Chan, I. Ahsan, V. Narayanan, S. R. Nandakumar, M. Le Gallo, P. A. Francese, <a href="https://research.ibm.com/people/abu-sebastian" rel="external nofollow noopener" target="_blank">A. Sebastian</a>, and E. Eleftheriou</div> <div class="periodical"> <em>In 2021 Symposium on VLSI Technology</em>, 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://ieeexplore.ieee.org/document/9508706" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> </div> <div class="badges"> <span class="altmetric-embed" data-altmetric-id="true" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>We present a 256x256 in-memory compute (IMC) core designed and fabricated in 14nm CMOS with backend-integrated multi-level phase-change memory (PCM). It comprises 256 linearized current controlled oscillator (CCO)-based ADCs at a compact 4µm pitch and a local digital processing unit performing affine scaling and ReLU operations. A novel frequency-linearization technique for CCOs is introduced, leading to accurate on-chip matrix-vector-multiply (MVM) when operating over 1 GHz. Measured classification accuracies on MNIST and CIFAR-10 datasets are presented when two cores are employed for deep learning (DL) inference. The measured energy efficiency is 10.5 TOPS/W at a performance density of 1.59 TOPS/mm 2 .</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">khaddam-aljameh_hermes_2021</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{{HERMES} {Core} – {A} 14nm {CMOS} and {PCM}-based {In}-{Memory} {Compute} {Core} using an array of 300ps/{LSB} {Linearized} {CCO}-based {ADCs} and local digital processing}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{2021 {Symposium} on {VLSI} {Technology}}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Khaddam-Aljameh, R. and Stanisavljevic, M. and Mas, J. Fornt and Karunaratne, G. and Braendli, M. and Liu, F. and Singh, A. and Müller, S. M. and Egger, U. and Petropoulos, A. and Antonakopoulos, T. and Brew, K. and Choi, S. and Ok, I. and Lie, F. L. and Saulnier, N. and Chan, V. and Ahsan, I. and Narayanan, V. and Nandakumar, S. R. and Gallo, M. Le and Francese, P. A. and Sebastian, A. and Eleftheriou, E.}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2021}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{1--2}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li></ol> <h2 class="year">2020</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">BIBM</abbr></div> <div id="lavenier_variant_2020" class="col-sm-8"> <div class="title">Variant Calling Parallelization on Processor-in-Memory Architecture</div> <div class="author"> <a href="https://lavenier.net/homepage/" rel="external nofollow noopener" target="_blank">D. Lavenier</a>, R. Cimadomo, and R. Jodin</div> <div class="periodical"> <em>In 2020 IEEE International Conference on Bioinformatics and Biomedicine (BIBM)</em>, Dec 2020 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.ieeecomputersociety.org/10.1109/BIBM49941.2020.9313351" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1109/BIBM49941.2020.9313351" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1109/BIBM49941.2020.9313351" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>This paper introduces a new combination of software and hardware PIM (Process-in-Memory) architecture to accelerate the variant calling genomic process. PIM translates into bringing data intensive calculations directly where the data is: within the DRAM, enhanced with thousands of processing units. The energy consumption, in large part due to data movement, is significantly lowered at a marginal additional hardware cost. Such design allows an unprecedented level of parallelism to process billions of short reads. Experiments on real PIM devices developed by the UPMEM company show significant speed-up compared to pure software implementation. The PIM solution also compared nicely to FPGA or GPU based acceleration bringing similar to twice the processing speed but most importantly being 5 to 8 times cheaper to deploy with up to 6 times less power consumption.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">lavenier_variant_2020</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">{Los Alamitos, CA, USA}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Variant {Calling} {Parallelization} on {Processor}-in-{Memory} {Architecture}}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://doi.ieeecomputersociety.org/10.1109/BIBM49941.2020.9313351}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1109/BIBM49941.2020.9313351}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{2020 {IEEE} {International} {Conference} on {Bioinformatics} and {Biomedicine} ({BIBM})}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{IEEE Computer Society}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lavenier, D. and Cimadomo, R. and Jodin, R.}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">dec</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2020}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{204--207}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">MICRO</abbr></div> <div id="cali_genasm_2020" class="col-sm-8"> <div class="title">GenASM: A High-Performance, Low-Power Approximate String Matching Acceleration Framework for Genome Sequence Analysis</div> <div class="author"> <a href="https://damlasenolcali.github.io" rel="external nofollow noopener" target="_blank">Damla Senol Cali</a>, <a href="https://in.linkedin.com/in/gurpreet-singh-kalsi-bb75302" rel="external nofollow noopener" target="_blank">Gurpreet S. Kalsi</a>, <a href="http://www.alkanlab.org/team/z%C3%BClal-bing%C3%B6l/" rel="external nofollow noopener" target="_blank">Zülal Bingöl</a>, <a href="https://cfirtina.com" rel="external nofollow noopener" target="_blank">Can Firtina</a>, Lavanya Subramanian, <a href="https://scholar.google.com/citations?user=-pOH1s4AAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Jeremie S. Kim</a>, Rachata Ausavarungnirun, <a href="https://mealser.github.io" rel="external nofollow noopener" target="_blank">Mohammed Alser</a>, <a href="https://ee.ethz.ch/the-department/people-a-z/person-detail.MjQxMjgy.TGlzdC8zMjc5LC0xNjUwNTg5ODIw.html" rel="external nofollow noopener" target="_blank">Juan Gomez-Luna</a>, Amirali Boroumand, Anant Norion, Allison Scibisz, <a href="https://in.linkedin.com/in/sreenivas-subramoney-i-am-hiring-6564521" rel="external nofollow noopener" target="_blank">Sreenivas Subramoneyon</a>, <a href="https://www.alkanlab.org/team/can-alkan/" rel="external nofollow noopener" target="_blank">Can Alkan</a>, <a href="https://ghose.web.illinois.edu" rel="external nofollow noopener" target="_blank">Saugata Ghose</a>, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em>In 2020 53rd Annual IEEE/ACM International Symposium on Microarchitecture (MICRO)</em>, Dec 2020 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.org/10.1109/MICRO50266.2020.00081" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://people.inf.ethz.ch/omutlu/pub/GenASM-approximate-string-matching-framework-for-genome-analysis_micro20.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> <a href="https://github.com/CMU-SAFARI/GenASM" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> <a href="https://people.inf.ethz.ch/omutlu/pub/GenASM-approximate-string-matching-framework-for-genome-analysis_micro20-talk.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Slides</a> <a href="https://people.inf.ethz.ch/omutlu/pub/GenASM-approximate-string-matching-framework-for-genome-analysis_micro20-talk.pptx" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Slides (PPT)</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1109/MICRO50266.2020.00081" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1109/MICRO50266.2020.00081" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Genome sequence analysis has enabled significant advancements in medical and scientific areas such as personalized medicine, outbreak tracing, and the understanding of evolution. To perform genome sequencing, devices extract small random fragments of an organism’s DNA sequence (known as reads). The first step of genome sequence analysis is a computational process known as read mapping. In read mapping, each fragment is matched to its potential location in the reference genome with the goal of identifying the original location of each read in the genome. Unfortunately, rapid genome sequencing is currently bottlenecked by the computational power and memory bandwidth limitations of existing systems, as many of the steps in genome sequence analysis must process a large amount of data. A major contributor to this bottleneck is approximate string matching (ASM), which is used at multiple points during the mapping process. ASM enables read mapping to account for sequencing errors and genetic variations in the reads. We propose GenASM, the first ASM acceleration framework for genome sequence analysis. GenASM performs bitvectorbased ASM, which can efficiently accelerate multiple steps of genome sequence analysis. We modify the underlying ASM algorithm (Bitap) to significantly increase its parallelism and reduce its memory footprint. Using this modified algorithm, we design the first hardware accelerator for Bitap. Our hardware accelerator consists of specialized systolic-array-based compute units and on-chip SRAMs that are designed to match the rate of computation with memory capacity and bandwidth, resulting in an efficient design whose performance scales linearly as we increase the number of compute units working in parallel. We demonstrate that GenASM provides significant performance and power benefits for three different use cases in genome sequence analysis. First, GenASM accelerates read alignment for both long reads and short reads. For long reads, GenASM outperforms state-of-the-art software and hardware accelerators by 116x and 3.9x, respectively, while reducing power consumption by 37x and 2.7x. For short reads, GenASM outperforms state-of-the-art software and hardware accelerators by 111x and 1.9x. Second, GenASM accelerates pre-alignment filtering for short reads, with 3.7x the performance of a state-of-the-art pre-alignment filter, while reducing power consumption by 1.7x and significantly improving the filtering accuracy. Third, GenASM accelerates edit distance calculation, with 22-12501x and 9.3-400x speedups over the state-of-the-art software library and FPGA-based accelerator, respectively, while reducing power consumption by 548-582x and 67x. We conclude that GenASM is a flexible, high-performance, and low-power framework, and we briefly discuss four other use cases that can benefit from GenASM.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">cali_genasm_2020</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{{GenASM}: {A} {High}-{Performance}, {Low}-{Power} {Approximate} {String} {Matching} {Acceleration} {Framework} for {Genome} {Sequence} {Analysis}}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1109/MICRO50266.2020.00081}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{2020 53rd {Annual} {IEEE}/{ACM} {International} {Symposium} on {Microarchitecture} ({MICRO})}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Cali, Damla Senol and Kalsi, Gurpreet S. and Bingöl, Zülal and Firtina, Can and Subramanian, Lavanya and Kim, Jeremie S. and Ausavarungnirun, Rachata and Alser, Mohammed and Gomez-Luna, Juan and Boroumand, Amirali and Norion, Anant and Scibisz, Allison and Subramoneyon, Sreenivas and Alkan, Can and Ghose, Saugata and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2020}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{951--966}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">SYSTOR</abbr></div> <div id="kaplan_bioseal_2020" class="col-sm-8"> <div class="title">BioSEAL: In-Memory Biological Sequence Alignment Accelerator for Large-Scale Genomic Data</div> <div class="author"> Roman Kaplan, <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a>, and <a href="https://webee.technion.ac.il/~ran/" rel="external nofollow noopener" target="_blank">Ran Ginosar</a> </div> <div class="periodical"> <em>In Proceedings of the 13th ACM International Systems and Storage Conference</em>, Dec 2020 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.org/10.1145/3383669.3398279" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1145/3383669.3398279" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1145/3383669.3398279" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Genome sequences contain hundreds of millions of DNA base pairs. Finding the degree of similarity between two genomes requires executing a compute-intensive dynamic programming algorithm, such as Smith-Waterman. Traditional von Neumann architectures have limited parallelism and cannot provide an efficient solution for large-scale genomic data. Approximate heuristic methods (e.g. BLAST) are commonly used. However, they are suboptimal and still compute-intensive.In this work, we present BioSEAL, a biological sequence alignment accelerator. BioSEAL is a massively parallel non-von Neumann processing-in-memory architecture for large-scale DNA and protein sequence alignment. BioSEAL is based on resistive content addressable memory, capable of energy-efficient and highperformance associative processing.We present an associative processing algorithm for entire database sequence alignment on BioSEAL and compare its performance and power consumption with state-of-art solutions. We show that BioSEAL can achieve up to 57x speedup and 156x better energy efficiency, compared with existing solutions for genome sequence alignment and protein sequence database search.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">kaplan_bioseal_2020</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">{New York, NY, USA}</span><span class="p">,</span>
  <span class="na">series</span> <span class="p">=</span> <span class="s">{{SYSTOR} '20}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{{BioSEAL}: {In}-{Memory} {Biological} {Sequence} {Alignment} {Accelerator} for {Large}-{Scale} {Genomic} {Data}}</span><span class="p">,</span>
  <span class="na">isbn</span> <span class="p">=</span> <span class="s">{978-1-4503-7588-7}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://doi.org/10.1145/3383669.3398279}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1145/3383669.3398279}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 13th {ACM} {International} {Systems} and {Storage} {Conference}}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Association for Computing Machinery}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Kaplan, Roman and Yavits, Leonid and Ginosar, Ran}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2020}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{36--48}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> </ol> <h2 class="year">2019</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">IEEE Micro</abbr></div> <div id="kaplan_rassa_2019" class="col-sm-8"> <div class="title">RASSA: Resistive Prealignment Accelerator for Approximate DNA Long Read Mapping</div> <div class="author"> Roman Kaplan, <a href="https://scholar.google.co.il/citations?user=ARyufyQAAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Leonid Yavits</a>, and <a href="https://webee.technion.ac.il/~ran/" rel="external nofollow noopener" target="_blank">Ran Ginosar</a> </div> <div class="periodical"> <em>IEEE Micro</em>, Jul 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.org/10.1109/MM.2018.2890253" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1109/MM.2018.2890253" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1109/MM.2018.2890253" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>DNA read mapping is a computationally expensive bioinformatics task, required for genome assembly and consensus polishing. It requires to find the best-fitting location for each DNA read on a long reference sequence. A novel resistive approximate similarity search accelerator (RASSA) exploits charge distribution and parallel in-memory processing to reflect a mismatch count between DNA sequences. RASSA implementation of DNA long-read prealignment outperforms the state-of-the-art solution, minimap2, by 16–77x with comparable accuracy and provides two orders of magnitude higher throughput than GateKeeper, a short-read prealignment hardware architecture implemented in FPGA.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">kaplan_rassa_2019</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{{RASSA}: {Resistive} {Prealignment} {Accelerator} for {Approximate} {DNA} {Long} {Read} {Mapping}}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{39}</span><span class="p">,</span>
  <span class="na">issn</span> <span class="p">=</span> <span class="s">{0272-1732}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://doi.org/10.1109/MM.2018.2890253}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1109/MM.2018.2890253}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{4}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{IEEE Micro}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Kaplan, Roman and Yavits, Leonid and Ginosar, Ran}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jul</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2019}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{44--54}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li></ol> <h2 class="year">2018</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">BMC Genomics</abbr></div> <div id="kim_grim-filter_2018" class="col-sm-8"> <div class="title">GRIM-Filter: Fast seed location filtering in DNA read mapping using processing-in-memory technologies</div> <div class="author"> <a href="https://scholar.google.com/citations?user=-pOH1s4AAAAJ&amp;hl=en" rel="external nofollow noopener" target="_blank">Jeremie S. Kim</a>, <a href="https://damlasenolcali.github.io" rel="external nofollow noopener" target="_blank">Damla Senol Cali</a>, Hongyi Xin, Donghyuk Lee, <a href="https://ghose.web.illinois.edu" rel="external nofollow noopener" target="_blank">Saugata Ghose</a>, <a href="https://mealser.github.io" rel="external nofollow noopener" target="_blank">Mohammed Alser</a>, Hasan Hassan, Oguz Ergin, <a href="https://www.alkanlab.org/team/can-alkan/" rel="external nofollow noopener" target="_blank">Can Alkan</a>, and <a href="https://people.inf.ethz.ch/omutlu/" rel="external nofollow noopener" target="_blank">Onur Mutlu</a> </div> <div class="periodical"> <em>BMC Genomics</em>, May 2018 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.org/10.1186/s12864-018-4460-0" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> <a href="https://bmcgenomics.biomedcentral.com/counter/pdf/10.1186/s12864-018-4460-0.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">PDF</a> <a href="https://github.com/CMU-SAFARI/GRIM" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> <a href="https://people.inf.ethz.ch/omutlu/pub/GRIM-filter-DNA-pre-alignment-in-memory_apbc18-talk.pdf" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Slides</a> <a href="https://people.inf.ethz.ch/omutlu/pub/GRIM-filter-DNA-pre-alignment-in-memory_apbc18-talk.pptx" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Slides (PPT)</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1186/s12864-018-4460-0" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1186/s12864-018-4460-0" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Seed location filtering is critical in DNA read mapping, a process where billions of DNA fragments (reads) sampled from a donor are mapped onto a reference genome to identify genomic variants of the donor. State-of-the-art read mappers 1) quickly generate possible mapping locations for seeds (i.e., smaller segments) within each read, 2) extract reference sequences at each of the mapping locations, and 3) check similarity between each read and its associated reference sequences with a computationally-expensive algorithm (i.e., sequence alignment) to determine the origin of the read. A seed location filter comes into play before alignment, discarding seed locations that alignment would deem a poor match. The ideal seed location filter would discard all poor match locations prior to alignment such that there is no wasted computation on unnecessary alignments.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">kim_grim-filter_2018</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{{GRIM}-{Filter}: {Fast} seed location filtering in {DNA} read mapping using processing-in-memory technologies}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{19}</span><span class="p">,</span>
  <span class="na">issn</span> <span class="p">=</span> <span class="s">{1471-2164}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://doi.org/10.1186/s12864-018-4460-0}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1186/s12864-018-4460-0}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{2}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{BMC Genomics}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Kim, Jeremie S. and Senol Cali, Damla and Xin, Hongyi and Lee, Donghyuk and Ghose, Saugata and Alser, Mohammed and Hassan, Hasan and Ergin, Oguz and Alkan, Can and Mutlu, Onur}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">may</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2018}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{89}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li></ol> <h2 class="year">2017</h2> <ol class="bibliography"></ol> <h2 class="year">2016</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 abbr"><abbr class="badge">BIBM</abbr></div> <div id="lavenier_dna_2016" class="col-sm-8"> <div class="title">DNA mapping using Processor-in-Memory architecture</div> <div class="author"> <a href="https://lavenier.net/homepage/" rel="external nofollow noopener" target="_blank">Dominique Lavenier</a>, Jean-Francois Roy, and David Furodet</div> <div class="periodical"> <em>In 2016 IEEE International Conference on Bioinformatics and Biomedicine (BIBM)</em>, May 2016 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abstract</a> <a class="bibtex btn btn-sm z-depth-0" role="button">BIB</a> <a href="https://doi.org/10.1109/BIBM.2016.7822732" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">URL</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1109/BIBM.2016.7822732" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1109/BIBM.2016.7822732" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>This paper presents the implementation of a mapping algorithm on a new Processing-in-Memory (PIM) architecture developed by UPMEM Company. UPMEM’s solution consists in adding processing units into the DRAM, to minimize data access time and maximize bandwidth, in order to drastically accelerate data-consuming algorithms. The technology developed by UPMEM makes it possible to combine 256 cores with 16 GBytes of DRAM, on a standard DIMM module. An experimentation of DNA Mapping on Human genome dataset shows that a speed-up of 25 can be obtained with UPMEM technology compared to fast mapping software such as BWA, Bowtie2 or NextGenMap running on 16 Intel threads. Experimentation also highlight that data transfer from storage device limits the performances of the implementation. The use of SSD drives can boost the speed-up to 80.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">lavenier_dna_2016</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{{DNA} mapping using {Processor}-in-{Memory} architecture}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1109/BIBM.2016.7822732}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{2016 {IEEE} {International} {Conference} on {Bioinformatics} and {Biomedicine} ({BIBM})}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Lavenier, Dominique and Roy, Jean-Francois and Furodet, David}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2016}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{1429--1435}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li></ol> </div> </article> </div> </div> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js" integrity="sha256-fgLAgv7fyCGopR/gBNq2iW3ZKIdqIcyshnUULC4vex8=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="/assets/js/common.js"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script async src="https://www.googletagmanager.com/gtag/js?id=G-GT1JB16ZYG"></script> <script>function gtag(){window.dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-GT1JB16ZYG");</script> <script async src="https://cdn.panelbear.com/analytics.js?site="></script> <script>window.panelbear=window.panelbear||function(){(window.panelbear.q=window.panelbear.q||[]).push(arguments)},panelbear("config",{site:""});</script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>